---
output: html_document
editor_options: 
  chunk_output_type: console
---
Null-W
===

Script by   
Sam Yeaman (1)  
Kathryn Hodgins (1,2)  
Lotte van Boheemen (2)  

1) Original script, Yeaman & Hodgins et al., 2016
2) Amendments to apply to ragweed dataset

```{r,include=FALSE}
library("tidyverse")
```

```{r}
top=0.99
```

Step 1: NA versus EU: Get all EU and NA SNPs in non-superoutlier windows for both regions
```{r}
snp_allwindow_na <- read.table("xtxna_1000-1000_top99_50miss_allwinds_SNPinfo",header = T)
snp_allwindow_na$windID <- paste(snp_allwindow_na$contig, snp_allwindow_na$win.start, snp_allwindow_na$win.end, sep="__") #all SNPs and window information they were allocated to
allwind_na <- read.table("sres1000-1000_na99_50miss_allwinds", header = T) #all windows
allwind_na$windID  <- paste(allwind_na$contig, allwind_na$win.start, allwind_na$win.end, sep="__")
super_na <- subset(allwind_na, topcan_q=="TRUE") #only select outlier windows
backgroundSNP_na <- anti_join(snp_allwindow_na,super_na, by = "windID") # all SNPs (+info) NOT in a outlier window
nonout_win_na <- subset(allwind_na, topcan_q=="FALSE") #all non-outlier windows

# do the same for EU
snp_allwindow_eu <- read.table("xtxeu_1000-1000_top99_50miss_allwinds_SNPinfo",header = T)
snp_allwindow_eu$windID <- paste(snp_allwindow_eu$contig, snp_allwindow_eu$win.start, snp_allwindow_eu$win.end, sep="__") #all SNPs and window information they were allocated to
allwind_eu <- read.table("sres1000-1000_eu99_50miss_allwinds", header = T) #all outlier windows
allwind_eu$windID  <- paste(allwind_eu$contig, allwind_eu$win.start, allwind_eu$win.end, sep="__")
super_eu <- subset(allwind_eu, topcan_q=="TRUE") #only select outlier windows
backgroundSNP_eu <- anti_join(snp_allwindow_eu, super_eu, by = "windID") 
nonout_win_eu <- subset(allwind_eu, topcan_q=="FALSE")

# do the same for AU
snp_allwindow_au <- read.table("xtxau_1000-1000_top99_50miss_allwinds_SNPinfo",header = T)
snp_allwindow_au$windID <- paste(snp_allwindow_au$contig, snp_allwindow_au$win.start, snp_allwindow_au$win.end, sep="__") #all SNPs and window information they were allocated to
allwind_au <- read.table("sres1000-1000_au99_50miss_allwinds", header = T) #all outlier windows
allwind_au$windID  <- paste(allwind_au$contig, allwind_au$win.start, allwind_au$win.end, sep="__")
super_au <- subset(allwind_au, topcan_q=="TRUE") #only select outlier windows
backgroundSNP_au <- anti_join(snp_allwindow_au,super_au, by = "windID") 
nonout_win_au <- subset(allwind_au, topcan_q=="FALSE")

```


Step 2: Sample 1000 SNPs from these windows per range (as the XtX values are range specific)

Get all non-superoutliers that are also not NA without replacement 1k times

```{r}
set.seed (12345)
indsamp_na <- rowSums (is.na(backgroundSNP_na)) # counts the number of NA per row
which_na <- which (indsamp_na== 0) # select rows which have no NA                  
subsampled1K_na <- backgroundSNP_na[sample(which_na,10000,replace = F),]

set.seed (12346)
indsamp_eu <- rowSums (is.na(backgroundSNP_eu)) 
which_eu <- which (indsamp_eu== 0)                  
subsampled1K_eu <- backgroundSNP_eu[sample(which_eu,10000,replace = F),]

indsamp_au <- rowSums (is.na(backgroundSNP_au)) 
which_au <- which (indsamp_au== 0)                  
subsampled1K_au <- backgroundSNP_au[sample(which_au,10000,replace = F),]
```

Step 3: compare the xtx values of the 1000 random SNPs with the non-superoutlier XTX values for each window using a Wilcox test
```{r}
# this is the total number of unique non-outlier windows

nulldist1 <- array (NA,c(length(nonout_win_na$windID),2))

for (i in 1:length(nonout_win_na$windID)){
  ind1 <- which (backgroundSNP_na$windID == nonout_win_na$windID[i]) # loop through each non-outlier window to get the info (eg xtx)
  sub1 <- backgroundSNP_na[ind1, 3] # the third column here are the xtx values
  temp1 <- wilcox.test((subsampled1K_na$xtx), (sub1))
  nulldist1[i,2] <- temp1$statistic
  nulldist1[i,1] <- sum (is.na (sub1) == F)
}

nulldist2 <- nulldist1[which (is.na (nulldist1[,2]) == F),]
zscore_na <- (2 * nulldist2[,2] - 10000 * nulldist2[,1]) / sqrt ( 10000 * nulldist2[,1] * (10000 + nulldist2[,1] + 1)/ 3 )
```

Repeat for Europe
```{r}
nulldist1 <- array (NA,c(length(nonout_win_eu$windID),2))
for (i in 1:length(nonout_win_eu$windID)){
  ind1 <- which (backgroundSNP_eu$windID == nonout_win_eu$windID[i])
  sub1 <- backgroundSNP_eu[ind1, 3]
  temp1 <- wilcox.test((subsampled1K_eu$xtx), (sub1))
  nulldist1[i,2] <- temp1$statistic
  nulldist1[i,1] <- sum (is.na (sub1) == F)
}
nulldist2 <- nulldist1[which (is.na (nulldist1[,2]) == F),]
zscore_eu <- (2 * nulldist2[,2] - 10000 * nulldist2[,1]) / sqrt ( 10000 * nulldist2[,1] * (10000 + nulldist2[,1] + 1)/ 3 )
```

And Australia
```{r}
nulldist1 <- array (NA,c(length(nonout_win_au$windID),2))
for (i in 1:length(nonout_win_au$windID)){
  ind1 <- which (backgroundSNP_au$windID == nonout_win_au$windID[i])
  sub1 <- backgroundSNP_au[ind1, 3]
  temp1 <- wilcox.test(subsampled1K_au$xtx, sub1)
  #temp2 <- wilcox.test((subsampled1K_au$xtx),(sub1))
  nulldist1[i,2] <- temp1$statistic
  nulldist1[i,1] <- sum (is.na (sub1) == F)
}
nulldist2 <- nulldist1[which (is.na (nulldist1[,2]) == F),]
zscore_au <- (2 * nulldist2[,2] - 10000 * nulldist2[,1]) / sqrt ( 10000 * nulldist2[,1] * (10000 + nulldist2[,1] + 1)/ 3 )
```

```{r}
write.table(zscore_na, "zscore_na_50miss")
write.table(zscore_eu, "zscore_eu_50miss")
write.table(zscore_au, "zscore_au_50miss")

#zscore_na = read.table("zscore_na")
#zscore_eu = read.table("zscore_eu")
#zscore_au = read.table("zscore_au")
```


Step 5: find outlier window in one range in the second range and test these windows against the null SNP set and then compare these W values to the null distribution

Find signatures of selection in EU windows identified as outliers in NA
```{r}
sub_super_na <- unique(as.character(super_na$windID))

count_loop2 <- 0

results_1 <- array (NA, c(length(sub_super_na),10))
#results_1 <- data.frame(window = numeric(), p = numeric(), stat = numeric(), n = numeric(), mean_z_full = numeric (), mean_z_1K = numeric(), c= numeric(), d= numeric())

if (length (sub_super_na) > 0){ #only start loop if we have outlier-windows
  
  for (zz in 1:length (sub_super_na)){
    
    samesame_butdifferent <- snp_allwindow_eu[snp_allwindow_eu$windID == sub_super_na[zz],3]
    # find xtx values of all SNPs of windows (both outlier and non-outlier) in one range that were outlier in the other
    rank <- snp_allwindow_eu[snp_allwindow_eu$windID == sub_super_na[zz],4]
    num.outlier <- sum(rank >= top) # total number of outlier SNPs in window
    
    testn <- sum (is.na(samesame_butdifferent) ==F) # total number of SNPs in window
    testn2 <- sum (is.na (subsampled1K_eu[,3]) == F) #would this be anything else than 1K?
    
    count_loop2 <- count_loop2 + 1
    
    if (testn > 1){
      
      test1 <- wilcox.test ((samesame_butdifferent), (subsampled1K_eu[,3]))
      
      test1_zscore <- data.frame ("qt" = (2 * as.numeric (test1$statistic) - testn2 * testn) / sqrt ( testn2 * testn * (testn2 + testn + 1)/ 3 ))
 
      results_1[count_loop2,2] <- test1$p.value
      results_1[count_loop2,3] <- test1$statistic
      results_1[count_loop2,4] <- testn # number of SNPs in each window
      results_1[count_loop2,5] <- mean ((samesame_butdifferent),na.rm = T) 
      results_1[count_loop2,6] <- mean ((subsampled1K_eu[,3]), na.rm = T)
      results_1[count_loop2,7] <-	as.numeric (test1_zscore[1,1])
      results_1[count_loop2,8] <- 1 - (sum (as.numeric (test1_zscore[1,1]) > zscore_eu) / length (zscore_eu))
      results_1[count_loop2,9] <- num.outlier
      results_1[count_loop2,10] <- "outNAinEU"

    }
    results_1[count_loop2,1] <- sub_super_na[zz]
  
  } #zz
  
} ##if enough genes

results_1 
#write.table(results_1, "nullw_outNA_inEU.txt")

sub_super_eu <- unique(as.character(super_eu$windID))
count_loop2 <- 0
results_2 <- array (NA, c(length(sub_super_eu),10))
if (length (sub_super_eu) > 0){ #only start loop if we have outlier-windows
  for (zz in 1:length (sub_super_eu)){
    samesame_butdifferent <- snp_allwindow_na[snp_allwindow_na$windID == sub_super_eu[zz],3]
    rank <- snp_allwindow_na[snp_allwindow_na$windID == sub_super_eu[zz],4]
    num.outlier <- sum(rank >= top) # total number of outlier SNPs in window
    testn <- sum (is.na(samesame_butdifferent) ==F) # total number of SNPs in window
    testn2 <- sum (is.na (subsampled1K_na[,3]) == F) #would this be anything else than 1K?
    count_loop2 <- count_loop2 + 1
    if (testn > 1){
      test1 <- wilcox.test ((samesame_butdifferent), (subsampled1K_na[,3]))
      test1_zscore <- data.frame ("qt" = (2 * as.numeric (test1$statistic) - testn2 * testn) / sqrt ( testn2 * testn * (testn2 + testn + 1)/ 3 ))
      results_2[count_loop2,2] <- test1$p.value
      results_2[count_loop2,3] <- test1$statistic
      results_2[count_loop2,4] <- testn # number of SNPs in each window
      results_2[count_loop2,5] <- mean ((samesame_butdifferent),na.rm = T) 
      results_2[count_loop2,6] <- mean ((subsampled1K_na[,3]), na.rm = T)
      results_2[count_loop2,7] <-	as.numeric (test1_zscore[1,1])
      results_2[count_loop2,8] <- 1 - (sum (as.numeric (test1_zscore[1,1]) > zscore_na) / length (zscore_na))
      results_2[count_loop2,9] <- num.outlier
      results_2[count_loop2,10] <- "outEUinNA"
  }
    results_2[count_loop2,1] <- sub_super_eu[zz]
  } #zz
} ##if enough genes
results_2
#write.table(results_2, "nullw_outEU_inNA.txt")

sub_super_na <- unique(as.character(super_na$windID))
count_loop2 <- 0
results_3 <- array (NA, c(length(sub_super_na),10))
if (length (sub_super_na) > 0){ #only start loop if we have outlier-windows
  for (zz in 1:length (sub_super_na)){
    samesame_butdifferent <- snp_allwindow_au[snp_allwindow_au$windID == sub_super_na[zz],3]
      rank <- snp_allwindow_au[snp_allwindow_au$windID == sub_super_na[zz],4]
    num.outlier <- sum(rank >= top) # total number of outlier SNPs in window
    testn <- sum (is.na(samesame_butdifferent) ==F) # total number of SNPs in window
    testn2 <- sum (is.na (subsampled1K_au[,3]) == F) #would this be anything else than 1K?
    count_loop2 <- count_loop2 + 1
    if (testn > 1){
      test1 <- wilcox.test ((samesame_butdifferent), (subsampled1K_au[,3]))
      test1_zscore <- data.frame ("qt" = (2 * as.numeric (test1$statistic) - testn2 * testn) / sqrt ( testn2 * testn * (testn2 + testn + 1)/ 3 ))
      results_3[count_loop2,2] <- test1$p.value
      results_3[count_loop2,3] <- test1$statistic
      results_3[count_loop2,4] <- testn # number of SNPs in each window
      results_3[count_loop2,5] <- mean ((samesame_butdifferent),na.rm = T) 
      results_3[count_loop2,6] <- mean ((subsampled1K_au[,3]), na.rm = T)
      results_3[count_loop2,7] <-	as.numeric (test1_zscore[1,1])
      results_3[count_loop2,8] <- 1 - (sum (as.numeric (test1_zscore[1,1]) > zscore_au) / length (zscore_au))
      results_3[count_loop2,9] <- num.outlier
      results_3[count_loop2,10] <- "outNAinAU"

    }
    results_3[count_loop2,1] <- sub_super_na[zz]
  } #zz
} ##if enough genes
results_3
#write.table(results_3, "nullw_outNA_inAU.txt")

sub_super_eu <- unique(as.character(super_eu$windID))
count_loop2 <- 0
results_4 <- array (NA, c(length(sub_super_eu),10))
if (length (sub_super_na) > 0){ #only start loop if we have outlier-windows
  for (zz in 1:length (sub_super_eu)){
    samesame_butdifferent <- snp_allwindow_au[snp_allwindow_au$windID == sub_super_eu[zz],3]
      rank <- snp_allwindow_au[snp_allwindow_au$windID == sub_super_eu[zz],4]
    num.outlier <- sum(rank >= top) # total number of outlier SNPs in window
    testn <- sum (is.na(samesame_butdifferent) ==F) # total number of SNPs in window
    testn2 <- sum (is.na (subsampled1K_au[,3]) == F) #would this be anything else than 1K?
    count_loop2 <- count_loop2 + 1
    if (testn > 1){
      test1 <- wilcox.test ((samesame_butdifferent), (subsampled1K_au[,3]))
      test1_zscore <- data.frame ("qt" = (2 * as.numeric (test1$statistic) - testn2 * testn) / sqrt ( testn2 * testn * (testn2 + testn + 1)/ 3 ))
      results_4[count_loop2,2] <- test1$p.value
      results_4[count_loop2,3] <- test1$statistic
      results_4[count_loop2,4] <- testn # number of SNPs in each window
      results_4[count_loop2,5] <- mean ((samesame_butdifferent),na.rm = T) 
      results_4[count_loop2,6] <- mean ((subsampled1K_au[,3]), na.rm = T)
      results_4[count_loop2,7] <-	as.numeric (test1_zscore[1,1])
      results_4[count_loop2,8] <- 1 - (sum (as.numeric (test1_zscore[1,1]) > zscore_au) / length (zscore_au))
      results_4[count_loop2,9] <- num.outlier
      results_4[count_loop2,10] <- "outEUinAU"

    }
    results_4[count_loop2,1] <- sub_super_eu[zz]
  } #zz
} ##if enough genes
results_4
#write.table(results_4, "nullw_outEU_inAU.txt")

sub_super_au <- unique(as.character(super_au$windID))
count_loop2 <- 0
results_5 <- array (NA, c(length(sub_super_au),10))
if (length (sub_super_au) > 0){ #only start loop if we have outlier-windows
  for (zz in 1:length (sub_super_au)){
    samesame_butdifferent <- snp_allwindow_na[snp_allwindow_na$windID == sub_super_au[zz],3]
    rank <- snp_allwindow_na[snp_allwindow_na$windID == sub_super_au[zz],4]
    num.outlier <- sum(rank >= top) # total number of outlier SNPs in window
    testn <- sum (is.na(samesame_butdifferent) ==F) # total number of SNPs in window
    testn2 <- sum (is.na (subsampled1K_na[,3]) == F) #would this be anything else than 1K?
    count_loop2 <- count_loop2 + 1
    if (testn > 1){
      test1 <- wilcox.test ((samesame_butdifferent), (subsampled1K_na[,3]))
      test1_zscore <- data.frame ("qt" = (2 * as.numeric (test1$statistic) - testn2 * testn) / sqrt ( testn2 * testn * (testn2 + testn + 1)/ 3 ))
      results_5[count_loop2,2] <- test1$p.value
      results_5[count_loop2,3] <- test1$statistic
      results_5[count_loop2,4] <- testn # number of SNPs in each window
      results_5[count_loop2,5] <- mean ((samesame_butdifferent),na.rm = T) 
      results_5[count_loop2,6] <- mean ((subsampled1K_na[,3]), na.rm = T)
      results_5[count_loop2,7] <-	as.numeric (test1_zscore[1,1])
      results_5[count_loop2,8] <- 1 - (sum (as.numeric (test1_zscore[1,1]) > zscore_na) / length (zscore_na))
      results_5[count_loop2,9] <- num.outlier
      results_5[count_loop2,10] <- "outAUinNA"

    }
    results_5[count_loop2,1] <- sub_super_au[zz]
  } #zz
} ##if enough genes
results_5
#write.table(results_5, "nullw_outAU_inNA.txt")

sub_super_au <- unique(as.character(super_au$windID))
count_loop2 <- 0
results_6 <- array (NA, c(length(sub_super_au),10))
if (length (sub_super_au) > 0){ #only start loop if we have outlier-windows
  for (zz in 1:length (sub_super_au)){
    samesame_butdifferent <- snp_allwindow_eu[snp_allwindow_eu$windID == sub_super_au[zz],3]
    rank <- snp_allwindow_eu[snp_allwindow_eu$windID == sub_super_au[zz],4]
    num.outlier <- sum(rank >= top) # total number of outlier SNPs in window
    testn <- sum (is.na(samesame_butdifferent) ==F) # total number of SNPs in window
    testn2 <- sum (is.na (subsampled1K_eu[,3]) == F) #would this be anything else than 1K?
    count_loop2 <- count_loop2 + 1
    if (testn > 1){
      test1 <- wilcox.test ((samesame_butdifferent), (subsampled1K_eu[,3]))
      test1_zscore <- data.frame ("qt" = (2 * as.numeric (test1$statistic) - testn2 * testn) / sqrt ( testn2 * testn * (testn2 + testn + 1)/ 3 ))
      results_6[count_loop2,2] <- test1$p.value
      results_6[count_loop2,3] <- test1$statistic
      results_6[count_loop2,4] <- testn # number of SNPs in each window
      results_6[count_loop2,5] <- mean ((samesame_butdifferent),na.rm = T) 
      results_6[count_loop2,6] <- mean ((subsampled1K_eu[,3]), na.rm = T)
      results_6[count_loop2,7] <-	as.numeric (test1_zscore[1,1])
      results_6[count_loop2,8] <- 1 - (sum (as.numeric (test1_zscore[1,1]) > zscore_eu) / length (zscore_eu))
      results_6[count_loop2,9] <- num.outlier
      results_6[count_loop2,10] <- "outAUinEU"

    }
    results_6[count_loop2,1] <- sub_super_au[zz]
  } #zz
} ##if enough genes
results_6
#write.table(results_6, "nullw_outAU_inEU.txt")
```


```{r}
dat = rbind(results_1,results_2, results_3, results_4,results_5,results_6)

write.table(dat,"nullwxtx_1000_99_allwin_50miss_all10K.txt")
```


##END SCRIPT##