Environment-Allele Analysis
===

> Code by Lotte van Boheemen

The below code will do the following:
- Look at basic distribution of the data  
- Define what a significant association is between what environment and what allele --> EAAoutlier
- Calculate number of EAAoutliers based on sliding window, where window size is set to a certain number of bases
- Print out actual window size in bases (between first and last SNP in window)
- Print out window size in SNPs (important if window length is limiting factor)
- Print out number of EAAoutliers in SNPs
- Plot distribution of EAAoutliers and actual window sizes within each provided data set (ranges)

> Update 25-6-16: Code is streamlined to run as a function, plot most important graphs and output results, so that input data doesn't have to be changed inside the code
> Update June 2017: code is written to accomodate environmental variables over multiple columns


```{r,include=FALSE}
library("knitr")
library("dplyr")
library("plyr")
library("reshape")

opts_knit$set(root.dir = '/Users/lottevanboheemen/Documents/Monash/PhD/Analyses/Data')
```

---

**Read data**

Refgenome_data.txt is a file containing all the contigs of the reference genome, so also regions in which no SNPs are called
```{r}
cdata <- read.table("refgenome_data.txt",header=T)

# logbf_na contains log-transformed bf values averaged over log-transformed bf of 3 independent (different root) Bayenv2 runs using .5M iterations, for North-American sampling locations only
bfdat_alltmp <- read.table("bf_all_ddgbs.txt",header=T)
bfdat_all <- cbind(bfdat_alltmp[,1:2],log(bfdat_alltmp[,3:25]))
bfdat_na <- read.table("logbf_na_ddgbs.txt",header=T)
bfdat_eu <- read.table("logbf_eu_ddgbs.txt",header=T)
bfdat_au <- read.table("logbf_au_ddgbs.txt",header=T)
```


Add contig length information to logbf data by combining matrices {dplyr}
```{r}
bf_all <- left_join(bfdat_all,cdata,by="contig")
bf_all$contig <- as.factor(bf_all$contig)

bf_na <- left_join(bfdat_na,cdata,by="contig")
bf_na$contig <- as.factor(bf_na$contig)

bf_eu <- left_join(bfdat_eu, cdata,by="contig")
bf_eu$contig<-as.factor(bf_eu$contig)

bf_au <- left_join(bfdat_au, cdata,by="contig")
bf_au$contig <- as.factor(bf_au$contig)
```


---  

Outlier window where slide is based on position on the contig (so not SNP).

```{r}
mdata= melt(bf_all, id=c("contig", "location", "length"))
mdata$myrank = rank(mdata$value)/length(mdata$value)
mdata_lat = mdata[which(mdata$variable=="Latitude_all"),]
mdata_lon = mdata[which(mdata$variable=="Longitude_all"),]
mdata_bio4 = mdata[which(mdata$variable=="bio4_all"),]
mdata_bio17 = mdata[which(mdata$variable=="bio17_all"),]
mdata_bio15 = mdata[which(mdata$variable=="bio15_all"),]
mdata_bio12 = mdata[which(mdata$variable=="bio12_all"),]
mdata_bio1 = mdata[which(mdata$variable=="bio1_all"),]
mdata_bio2 = mdata[which(mdata$variable=="bio2_all"),]
mdata_bio8 = mdata[which(mdata$variable=="bio8_all"),]
mdata_alt = mdata[which(mdata$variable=="alt_all"),]

mdata2= melt(bf_na, id=c("contig", "location", "length"))
mdata2$myrank = rank(mdata2$value)/length(mdata2$value)
mdata2_lat = mdata2[which(mdata2$variable=="Latitude_all"),]
mdata2_lon = mdata2[which(mdata2$variable=="Longitude_all"),]
mdata2_bio4 = mdata2[which(mdata2$variable=="bio4_all"),]
mdata2_bio17 = mdata2[which(mdata2$variable=="bio17_all"),]
mdata2_bio15 = mdata2[which(mdata2$variable=="bio15_all"),]
mdata2_bio12 = mdata2[which(mdata2$variable=="bio12_all"),]
mdata2_bio1 = mdata2[which(mdata2$variable=="bio1_all"),]
mdata2_bio2 = mdata2[which(mdata2$variable=="bio2_all"),]
mdata2_bio8 = mdata2[which(mdata2$variable=="bio8_all"),]
mdata2_alt = mdata2[which(mdata2$variable=="alt_all"),]

mdata3= melt(bf_eu, id=c("contig", "location", "length"))
mdata3$myrank = rank(mdata3$value)/length(mdata3$value)
mdata3_lat = mdata3[which(mdata3$variable=="Latitude_all"),]
mdata3_lon = mdata3[which(mdata3$variable=="Longitude_all"),]
mdata3_bio4 = mdata3[which(mdata3$variable=="bio4_all"),]
mdata3_bio17 = mdata3[which(mdata3$variable=="bio17_all"),]
mdata3_bio15 = mdata3[which(mdata3$variable=="bio15_all"),]
mdata3_bio12 = mdata3[which(mdata3$variable=="bio12_all"),]
mdata3_bio1 = mdata3[which(mdata3$variable=="bio1_all"),]
mdata3_bio2 = mdata3[which(mdata3$variable=="bio2_all"),]
mdata3_bio8 = mdata3[which(mdata3$variable=="bio8_all"),]
mdata3_alt = mdata3[which(mdata3$variable=="alt_all"),]

mdata4= melt(bf_au, id=c("contig", "location", "length"))
mdata4$myrank = rank(mdata4$value)/length(mdata4$value)
mdata4_lat = mdata4[which(mdata4$variable=="Latitude_all"),]
mdata4_lon = mdata4[which(mdata4$variable=="Longitude_all"),]
mdata4_bio4 = mdata4[which(mdata4$variable=="bio4_all"),]
mdata4_bio17 = mdata4[which(mdata4$variable=="bio17_all"),]
mdata4_bio15 = mdata4[which(mdata4$variable=="bio15_all"),]
mdata4_bio12 = mdata4[which(mdata4$variable=="bio12_all"),]
mdata4_bio1 = mdata4[which(mdata4$variable=="bio1_all"),]
mdata4_bio2 = mdata4[which(mdata4$variable=="bio2_all"),]
mdata4_bio8 = mdata4[which(mdata4$variable=="bio8_all"),]
mdata4_alt = mdata4[which(mdata4$variable=="alt_all"),]

test = mdata2[c(1:1000,100000:101000,200000:201000),]
```

```{r}
#window length in bases
win.length <- 1000
# slide in bases
slide <- 1000
topBF <- 2

outlier.window <- function(data, output, top){
  
  ## run the main part of the outlier window function:

  # prepare the space for results to be written to
  results <- data.frame(envvar=numeric(),contig=numeric(),windID= numeric(),win.start=numeric(),win.end = numeric(), current.window = numeric(), num.SNPs = numeric(), num.outlier = numeric(), num.BFoutlier = numeric(), win.mean = numeric(), rank.mean= numeric() )
  
  SNPinfo <- data.frame(envvar=numeric(),contig=numeric(),location = numeric(), BF = numeric(), myrank = numeric (), win.start = numeric(), win.end = numeric())
  
  env <- levels(factor(data$variable))
  
  #set index for results, this is the index of window passed threshold
  j <- 1
  #set index for SNPinfo results, where every line is a SNP and all information
  a <- 1

  for(vv in 1:length(env)) {
  
  all_envdata = mdata[which(mdata$variable==env[vv]),]
  envdata = data[which(data$variable==env[vv]),]
  # create a vector including names of each contig for which SNPs have been found
  cont <- levels(factor(envdata$contig)) 
 
  # Loop through each contig, stop the loop at the final contig
  for(k in 1:length(cont)){
    
    data2 = envdata[envdata$contig==cont[k],]
    data2_all = all_envdata[all_envdata$contig==cont[k],]
    
     if(nrow(data2_all)>0){
    wID = 1 
    win.start = as.numeric(data2_all$location[[1]])
    last.loc <- as.numeric(max(data2$location))
    
    while(win.start <= last.loc){
   
      win.end = win.start + slide - 1
      wind = data2[which(data2$location>=win.start&data2$location<=win.end),]
      
      if(nrow(wind) > 0){

      # calculate the number of SNPs in the window
      numSNP <- as.numeric(nrow(wind))
      current.window = as.numeric(max(wind$location)- min(wind$location))
      num.outlier <- sum(wind$myrank >= top)
      num.BFoutlier = sum(wind$value >= topBF)
      win.mean <- mean(wind$value)
      rank.mean <- mean(wind$myrank)
      
      results[j, ] <- c(env[vv], cont[k], wID,win.start,win.end,current.window, numSNP, num.outlier,num.BFoutlier, win.mean, rank.mean)
      
      j = j + 1
      # create seperate file with info for every SNP

      for(b in 1:nrow(wind)){
        SNPinfo[a,] <- c(env[vv], cont[k], wind$location[b], wind$value[b],wind$myrank[b], win.start, win.end)
        a <- a + 1
      }
      }
      print(c(a,k, wID))
      win.start = win.start + slide
      wID = wID + 1
      
  }## window
      print(k)
     }
} ## contig
  } ##  vv

   write.table(results, file = output) 
   temp <- paste0(output,"_SNPinfo")
   write.table(SNPinfo, file = temp)
}
      
```

Call function for the different datasets
```{r}
outlier.window(mdata_lat, "bflat_all_1000_bf999",0.999)
outlier.window(mdata_lon, "bflon_all_1000_bf999",0.999)
outlier.window(mdata_alt, "bfalt_all_1000_bf999",0.999)
outlier.window(mdata_bio1, "bfbio1_all_1000_bf999",0.999)
outlier.window(mdata_bio12, "bfbio12_all_1000_bf999",0.999)
outlier.window(mdata_bio15, "bfbio15_all_1000_bf999",0.999)
outlier.window(mdata_bio17, "bfbio17_all_1000_bf999",0.999)
outlier.window(mdata_bio2, "bfbio2_all_1000_bf999",0.999)
outlier.window(mdata_bio4, "bfbio4_all_1000_bf999",0.999)
outlier.window(mdata_bio8, "bfbio8_all_1000_bf999",0.999)

outlier.window(mdata2_lat, "bflat_na_1000_bf999",0.999)
outlier.window(mdata2_lon, "bflon_na_1000_bf999",0.999)
outlier.window(mdata2_alt, "bfalt_na_1000_bf999",0.999)
outlier.window(mdata2_bio1, "bfbio1_na_1000_bf999",0.999)
outlier.window(mdata2_bio12, "bfbio12_na_1000_bf999",0.999)
outlier.window(mdata2_bio15, "bfbio15_na_1000_bf999",0.999)
outlier.window(mdata2_bio17, "bfbio17_na_1000_bf999",0.999)
outlier.window(mdata2_bio2, "bfbio2_na_1000_bf999",0.999)
outlier.window(mdata2_bio4, "bfbio4_na_1000_bf999",0.999)
outlier.window(mdata2_bio8, "bfbio8_na_1000_bf999",0.999)

outlier.window(mdata3_lat, "bflat_eu_1000_bf999",0.999)
outlier.window(mdata3_lon, "bflon_eu_1000_bf999",0.999)
outlier.window(mdata3_alt, "bfalt_eu_1000_bf999",0.999)
outlier.window(mdata3_bio1, "bfbio1_eu_1000_bf999",0.999)
outlier.window(mdata3_bio12, "bfbio12_eu_1000_bf999",0.999)
outlier.window(mdata3_bio15, "bfbio15_eu_1000_bf999",0.999)
outlier.window(mdata3_bio17, "bfbio17_eu_1000_bf999",0.999)
outlier.window(mdata3_bio2, "bfbio2_eu_1000_bf999",0.999)
outlier.window(mdata3_bio4, "bfbio4_eu_1000_bf999",0.999)
outlier.window(mdata3_bio8, "bfbio8_eu_1000_bf999",0.999)

outlier.window(mdata4_lat, "bflat_au_1000_bf999",0.999)
outlier.window(mdata4_lon, "bflon_au_1000_bf999",0.999)
outlier.window(mdata4_alt, "bfalt_au_1000_bf999",0.999)
outlier.window(mdata4_bio1, "bfbio1_au_1000_bf999",0.999)
outlier.window(mdata4_bio12, "bfbio12_au_1000_bf999",0.999)
outlier.window(mdata4_bio15, "bfbio15_au_1000_bf999",0.999)
outlier.window(mdata4_bio17, "bfbio17_au_1000_bf999",0.999)
outlier.window(mdata4_bio2, "bfbio2_au_1000_bf999",0.999)
outlier.window(mdata4_bio4, "bfbio4_au_1000_bf999",0.999)
outlier.window(mdata4_bio8, "bfbio8_au_1000_bf999",0.999)
```


##END SCRIPT##