---
output: html_document
editor_options: 
  chunk_output_type: console
---



```{r}
library("knitr")
library("plyr")
library("dplyr")
library("reshape")
library("ggplot2")
library("heatmap3")
library("cluster")
```

Read data
```{r}
wind_na = read.table("bf_1000_99_allwind/sresbfall_na_1000_99_allwind.txt", header=T)
wind_eu = read.table("bf_1000_99_allwind/sresbfall_eu_1000_99_allwind.txt", header=T)
wind_au = read.table("bf_1000_99_allwind/sresbfall_au_1000_99_allwind.txt", header=T)

xwind_na = read.table("xtx_1000_99_allwind/sres1000-1000_na99_allwinds", header=T)
xwind_eu = read.table("xtx_1000_99_allwind/sres1000-1000_eu99_allwinds", header=T)
xwind_au = read.table("xtx_1000_99_allwind/sres1000-1000_au99_allwinds", header=T)

xwind_na$windID = paste0(xwind_na$contig, "__", xwind_na$win.start)
xwind_eu$windID = paste0(xwind_eu$contig, "__", xwind_eu$win.start)
xwind_au$windID = paste0(xwind_au$contig, "__", xwind_au$win.start)

wind_na$windID = paste0(wind_na$contig, "__", wind_na$win.start)
wind_eu$windID = paste0(wind_eu$contig, "__", wind_eu$win.start)
wind_au$windID = paste0(wind_au$contig, "__", wind_au$win.start)
wind_na$windIDenv = paste0(wind_na$windID, "__", wind_na$envvar)
wind_eu$windIDenv = paste0(wind_eu$windID, "__", wind_eu$envvar)
wind_au$windIDenv = paste0(wind_au$windID, "__", wind_au$envvar)
#subset to binomial outlier windows

bfn_sup = read.table("bf_1000_99_allwind/mbf_1000_99_binnullW.txt", header=T)
bfxtx_sup = read.table("xtx_1000_99_allwind/mxtx_1000_99_binnullW.txt", header=T)

bfn_sup_NA = bfn_sup %>% filter(qbinom=="yes" & Range=="nat") %>% droplevels()
bfn_sup_EU = bfn_sup %>% filter(qbinom=="yes" & Range=="EU") %>% droplevels()
bfn_sup_AU = bfn_sup %>% filter(qbinom=="yes" & Range=="AU") %>% droplevels()

bfnw_sup_NA = bfn_sup %>% filter(null.W=="yes" & Range=="nat") %>% droplevels()
bfnw_sup_EU = bfn_sup %>% filter(null.W=="yes"& Range=="EU") %>% droplevels()
bfnw_sup_AU = bfn_sup %>% filter(null.W=="yes"& Range=="AU") %>% droplevels()

xn_sup_NA = bfxtx_sup %>% filter(binnullw=="yes" & Range=="nat") %>% droplevels()
xn_sup_EU = bfxtx_sup %>% filter(binnullw=="yes" & Range=="EU") %>% droplevels()
xn_sup_AU = bfxtx_sup %>% filter(binnullw=="yes" & Range=="AU") %>% droplevels()
```

Window has to have hit in at least 2 methods within a range (based on nullw)
---
```{r}
#with null-w results
bfnw_sup_NAall = as.data.frame(levels(as.factor(bfnw_sup_NA$windID)))
colnames(bfnw_sup_NAall) = "windID"
bfwxtx_sup_NA = inner_join(xn_sup_NA, bfnw_sup_NAall, by="windID")
bfnw_sup_EUall = as.data.frame(levels(as.factor(bfnw_sup_EU$windID)))
colnames(bfnw_sup_EUall) = "windID"
bfwxtx_sup_EU = inner_join(xn_sup_EU, bfnw_sup_EUall, by="windID")
bfnw_sup_AUall = as.data.frame(levels(as.factor(bfnw_sup_AU$windID)))
colnames(bfnw_sup_AUall) = "windID"
bfwxtx_sup_AU = inner_join(xn_sup_AU, bfnw_sup_AUall, by="windID")

bfwxtx_sup_NAEU = full_join(bfwxtx_sup_NA[,c(1,5)], bfwxtx_sup_EU[,c(1,5)], by="windID")
bfwxtx_sup_NAEU = full_join(bfwxtx_sup_NA[,c(1,5)], bfwxtx_sup_EU[,c(1,5)], by="windID")
bfwxtx_sup_NAEUAU = full_join(bfwxtx_sup_AU[,c(1,5)], bfwxtx_sup_NAEU, by="windID")
```


##NA
```{r}
rho_na <- read.table("rho_naddGBS", header=T)
rho_nam <- melt(rho_na, id=c("contig","location"))
rho_na$SNPID  <- paste(rho_na$contig, rho_na$location,sep="__")

SNP_na <- read.table("bf_1000_99_allwind/bfall_na_1000_bf99_allwinds_SNPinfo", header=T)
SNP_na$windID  <- paste0(SNP_na$contig, "__", SNP_na$win.start)

topSNP_na = semi_join(SNP_na, bfwxtx_sup_NA, by="windID")

topSNP_na$SNPID = paste(topSNP_na$contig, topSNP_na$location,sep="__")
```

###EU
```{r}
rho_eu <- read.table("rho_euddGBS", header=T)
rho_eum <- melt(rho_eu, id=c("contig","location"))
rho_eu$SNPID  <- paste(rho_eu$contig, rho_eu$location,sep="__")

SNP_eu <- read.table("bf_1000_99_allwind/bfall_eu_1000_bf99_allwinds_SNPinfo", header=T)
SNP_eu$windID  <- paste0(SNP_eu$contig, "__", SNP_eu$win.start)

topSNP_eu = semi_join(SNP_eu, bfwxtx_sup_EU, by="windID")

topSNP_eu$SNPID = paste(topSNP_eu$contig, topSNP_eu$location,sep="__")
```

##AU
```{r}
rho_au <- read.table("rho_auddGBS", header=T)
rho_aum <- melt(rho_au, id=c("contig","location"))
rho_au$SNPID  <- paste(rho_au$contig, rho_au$location,sep="__")

SNP_au <- read.table("bf_1000_99_allwind/bfall_au_1000_bf99_allwinds_SNPinfo", header=T)
SNP_au$windID  <- paste0(SNP_au$contig, "__", SNP_au$win.start)

topSNP_au = semi_join(SNP_au, bfwxtx_sup_AU, by="windID")

topSNP_au$SNPID = paste(topSNP_au$contig, topSNP_au$location,sep="__")
```


Rename
---
```{r}
rho_na$contig = gsub("SC2ABSE_","",rho_na$contig)
rho_eu$contig = gsub("SC2ABSE_","",rho_eu$contig)
rho_au$contig = gsub("SC2ABSE_","",rho_au$contig)
```

```{r}
colnames(rho_na) <- gsub("_all","",colnames(rho_na))
colnames(rho_na) <- gsub("E_","",colnames(rho_na))
colnames(rho_na) <- gsub("Latitudeabs","Lat",colnames(rho_na))
colnames(rho_na) <- gsub("Longitude","Long",colnames(rho_na))
colnames(rho_na) <- gsub("alt","Alt",colnames(rho_na))
colnames(rho_na) <- gsub("bio10","TQwarm",colnames(rho_na))
colnames(rho_na) <- gsub("bio11","TQcold",colnames(rho_na))
colnames(rho_na) <- gsub("bio12","Pmean",colnames(rho_na))
colnames(rho_na) <- gsub("bio13","Pmax",colnames(rho_na))
colnames(rho_na) <- gsub("bio14","Pmin",colnames(rho_na))
colnames(rho_na) <- gsub("bio15","Pseas",colnames(rho_na))
colnames(rho_na) <- gsub("bio16","PmaxQ",colnames(rho_na))
colnames(rho_na) <- gsub("bio17","PminQ",colnames(rho_na))
colnames(rho_na) <- gsub("bio18","PQwarm",colnames(rho_na))
colnames(rho_na) <- gsub("bio19","PQcold",colnames(rho_na))
colnames(rho_na) <- gsub("bio1","Tmean",colnames(rho_na))
colnames(rho_na) <- gsub("bio2","Tdiur",colnames(rho_na))
colnames(rho_na) <- gsub("bio3","Tiso",colnames(rho_na))
colnames(rho_na) <- gsub("bio4","Tseas",colnames(rho_na))
colnames(rho_na) <- gsub("bio5","Tmax",colnames(rho_na))
colnames(rho_na) <- gsub("bio6","Tmin",colnames(rho_na))
colnames(rho_na) <- gsub("bio7","Trange",colnames(rho_na))
colnames(rho_na) <- gsub("bio8","TQwet",colnames(rho_na))
colnames(rho_na) <- gsub("bio9","TQdry",colnames(rho_na))
```
```{r}
colnames(rho_eu) <- gsub("_all","",colnames(rho_eu))
colnames(rho_eu) <- gsub("E_","",colnames(rho_eu))
colnames(rho_eu) <- gsub("Latitudeabs","Lat",colnames(rho_eu))
colnames(rho_eu) <- gsub("Longitude","Long",colnames(rho_eu))
colnames(rho_eu) <- gsub("alt","Alt",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio10","TQwarm",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio11","TQcold",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio12","Pmean",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio13","Pmax",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio14","Pmin",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio15","Pseas",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio16","PmaxQ",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio17","PminQ",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio18","PQwarm",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio19","PQcold",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio1","Tmean",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio2","Tdiur",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio3","Tiso",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio4","Tseas",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio5","Tmax",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio6","Tmin",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio7","Trange",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio8","TQwet",colnames(rho_eu))
colnames(rho_eu) <- gsub("bio9","TQdry",colnames(rho_eu))
```
```{r}
colnames(rho_au) <- gsub("_all","",colnames(rho_au))
colnames(rho_au) <- gsub("E_","",colnames(rho_au))
colnames(rho_au) <- gsub("Latitudeabs","Lat",colnames(rho_au))
colnames(rho_au) <- gsub("Longitude","Long",colnames(rho_au))
colnames(rho_au) <- gsub("alt","Alt",colnames(rho_au))
colnames(rho_au) <- gsub("bio10","TQwarm",colnames(rho_au))
colnames(rho_au) <- gsub("bio11","TQcold",colnames(rho_au))
colnames(rho_au) <- gsub("bio12","Pmean",colnames(rho_au))
colnames(rho_au) <- gsub("bio13","Pmax",colnames(rho_au))
colnames(rho_au) <- gsub("bio14","Pmin",colnames(rho_au))
colnames(rho_au) <- gsub("bio15","Pseas",colnames(rho_au))
colnames(rho_au) <- gsub("bio16","PmaxQ",colnames(rho_au))
colnames(rho_au) <- gsub("bio17","PminQ",colnames(rho_au))
colnames(rho_au) <- gsub("bio18","PQwarm",colnames(rho_au))
colnames(rho_au) <- gsub("bio19","PQcold",colnames(rho_au))
colnames(rho_au) <- gsub("bio1","Tmean",colnames(rho_au))
colnames(rho_au) <- gsub("bio2","Tdiur",colnames(rho_au))
colnames(rho_au) <- gsub("bio3","Tiso",colnames(rho_au))
colnames(rho_au) <- gsub("bio4","Tseas",colnames(rho_au))
colnames(rho_au) <- gsub("bio5","Tmax",colnames(rho_au))
colnames(rho_au) <- gsub("bio6","Tmin",colnames(rho_au))
colnames(rho_au) <- gsub("bio7","Trange",colnames(rho_au))
colnames(rho_au) <- gsub("bio8","TQwet",colnames(rho_au))
colnames(rho_au) <- gsub("bio9","TQdry",colnames(rho_au))
```

Check if rho are comparable (and if coding of alleles in bayenv was the same)
```{r}
rho_naeu = inner_join(rho_na, rho_eu, by="SNPID")
plot(rho_naeu$Latitude.x~rho_naeu$Latitude.y)

rho_naau = inner_join(rho_na, rho_au, by="SNPID")
plot(rho_naau$Tmean.x~rho_naau$Tmean.y)
```

Get with random shared outlier SNPs
---

```{r}
#select outlier SNPs in outlier windows in >1 method with 90% SNP call rate
toptopSNP_NA =topSNP_na[ which(topSNP_na$rank>=0.99),] 
toptopSNP_EU =topSNP_eu[ which(topSNP_eu$rank>=0.99),]
toptopSNP_AU =topSNP_au[ which(topSNP_au$rank>=0.99),]

toptopSNP_NA_SNPs = dplyr::select(toptopSNP_NA, SNPID, windID)
toptopSNP_EU_SNPs = dplyr::select(toptopSNP_EU, SNPID)
toptopSNP_AU_SNPs = dplyr::select(toptopSNP_AU, SNPID)

topSNP_NAEU = inner_join(toptopSNP_NA_SNPs, toptopSNP_EU_SNPs, by="SNPID")
topSNP_NAEUAU = inner_join(topSNP_NAEU, toptopSNP_AU_SNPs, by="SNPID")

topSNP_NAEUAU_r <- ddply(topSNP_NAEUAU, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_NA_r <- ddply(toptopSNP_NA_SNPs, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
toptopSNP_EU_SNPs = dplyr::select(toptopSNP_EU, SNPID, windID)
toptopSNP_AU_SNPs = dplyr::select(toptopSNP_AU, SNPID, windID)
topSNP_EU_r <- ddply(toptopSNP_EU_SNPs, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_AU_r <- ddply(toptopSNP_AU_SNPs, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})

rho_natop <- semi_join(rho_na, topSNP_NAEUAU_r, by="SNPID")
rho_eutop <- semi_join(rho_eu, topSNP_NAEUAU_r, by="SNPID")
rho_autop <- semi_join(rho_au, topSNP_NAEUAU_r, by="SNPID")

rho_natop_sin <- semi_join(rho_na, topSNP_NA_r, by="SNPID")
rho_eutop_sin <- semi_join(rho_eu, topSNP_EU_r, by="SNPID")
rho_autop_sin <- semi_join(rho_au, topSNP_AU_r, by="SNPID")
```



Heatmaps for outliers in all ranges
---
NA
```{r}
dat <- (abs(rho_natop[,c(3:4, 6:25)]))  # numerical columns
rownames(dat) <- rho_natop[,1]
dat=t(dat)
a<- hclust(dist(dat), method="ward.D2")
b<-hclust(dist(t(dat)), method="ward.D2")
snpsplit_na_all = as.data.frame(cutree(b,4))
snpsplit_na_all$SNPID = rownames(snpsplit_na_all)
colnames(snpsplit_na_all) =c( "cluster","SNPID")

pdf("_Graph/heatmap_NAfull.99abs.pdf")
# Plot heatmap with heatmap.3
heatmap3(sqrt(dat),
          Colv=as.dendrogram(b),     # Reordered cols
          Rowv=as.dendrogram(a),     # Reordered rows
          method="ward.D2",
          scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 1.3,cexRow =1.3)
dev.off()


dat <- sqrt(abs(rho_eutop[,c(3:4, 6:25)]) )
rownames(dat) <- rho_eutop[,1]
dat=t(dat)
pdf("_Graph/heatmap_EUfull_NAord.99abs.pdf")
heatmap3(dat,Colv=as.dendrogram(b), Rowv=as.dendrogram(a),scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 1.3,cexRow =1.3)
dev.off()

dat <- sqrt(abs(rho_autop[,c(3:4, 6:25)]))
rownames(dat) <- rho_autop[,1]
dat=t(dat)
pdf("_Graph/heatmap_AUfull_NAord.99abs.pdf")
# Plot heatmap with heatmap.3
heatmap3(dat,Colv=as.dendrogram(b), Rowv=as.dendrogram(a),scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 1.3,cexRow =1.3)
dev.off()
```

EU
```{r}
dat <- abs(rho_eutop[,c(3:4, 6:25)])  # numerical columns
rownames(dat) <- rho_eutop[,1]
dat=t(dat)
b<-hclust(dist(t(dat)), method="ward.D2")
snpsplit_eu_all = as.data.frame(cutree(b,4))
snpsplit_eu_all$SNPID = rownames(snpsplit_eu_all)
colnames(snpsplit_eu_all) = c( "cluster","SNPID")

pdf("_Graph/heatmap_EUfull.99abs.pdf")
heatmap3(sqrt(dat),
          Colv=as.dendrogram(b),     # Reordered cols
          Rowv=as.dendrogram(a),     # Reordered rows
          method="ward.D2",
          scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 1.3,cexRow =1.3)
dev.off()
```

AU
```{r}
dat <- abs(rho_autop[,c(3:4, 6:25)])  # numerical columns
rownames(dat) <- rho_autop[,1]
dat=t(dat)
b<-hclust(dist(t(dat)), method="ward.D2")
snpsplit_au_all = as.data.frame(cutree(b,4))
snpsplit_au_all$SNPID = rownames(snpsplit_au_all)
colnames(snpsplit_au_all) = c( "cluster","SNPID")

pdf("_Graph/heatmap_AUfull.99abs.pdf")
heatmap3(sqrt(dat),
          Colv=as.dendrogram(b),     # Reordered cols
          Rowv=as.dendrogram(a),     # Reordered rows
          method="ward.D2",
          scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 1.3,cexRow =1.3)
dev.off()
```




Heatmap for per-range outliers
---

```{r}
rho_natop_sin <- ddply(rho_natop_sin, ~contig, function(x){ ndf <- x[sample(nrow(x),1), ]})
rho_eutop_sin <- ddply(rho_eutop_sin, ~contig, function(x){ ndf <- x[sample(nrow(x),1), ]})
rho_autop_sin <- ddply(rho_autop_sin, ~contig, function(x){ ndf <- x[sample(nrow(x),1), ]})
```

Write table to save the randomly picked SNPs
```{r}
#write.table(rho_natop_sin,"rho_natop_sin")
#write.table(rho_eutop_sin,"rho_eutop_sin")
#write.table(rho_autop_sin,"rho_autop_sin")

rho_natop_sin = read.table("rho_natop_sin", header=T)
rho_eutop_sin = read.table("rho_eutop_sin", header=T)
rho_autop_sin = read.table("rho_autop_sin", header=T)
```

NA
```{r}
dat <- (abs(rho_natop_sin[,c(3:4, 6:25)]))  # numerical columns
rownames(dat) <- rho_natop_sin[,1]
dat=t(dat)
a<- hclust(dist(dat), method="ward.D2")
b<-hclust(dist(t(dat)), method="ward.D2")

pdf("_Graph/heatmap_sinNAfull.99abs.pdf")
heatmap3(sqrt(dat),
          Colv=as.dendrogram(b),     # Reordered cols
          Rowv=as.dendrogram(a),     # Reordered rows
          method="ward.D2",
          scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 0.4,cexRow =1.3)
dev.off()

snpsplit_nasin_all = as.data.frame(cutree(b,4))
snpsplit_nasin_all$contig = rownames(snpsplit_nasin_all)
colnames(snpsplit_nasin_all) =c( "cluster","contig")
```


EU
```{r}
dat <- abs(rho_eutop_sin[,c(3:4, 6:25)])  # numerical columns
rownames(dat) <- rho_eutop_sin[,1]
dat=t(dat)
b<-hclust(dist(t(dat)), method="ward.D2")

pdf("_Graph/heatmap_sinEUfull.99abs.pdf")
heatmap3(sqrt(dat),
          Colv=as.dendrogram(b),     # Reordered cols
          Rowv=as.dendrogram(a),     # Reordered rows
          method="ward.D2",
          scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 0.4,cexRow =1.3)
dev.off()

snpsplit_eusin_all = as.data.frame(cutree(b,4))
snpsplit_eusin_all$contig = rownames(snpsplit_eusin_all)
colnames(snpsplit_eusin_all) = c( "cluster","contig")
```

AU
```{r}
dat <- abs(rho_autop_sin[,c(3:4, 6:25)])  # numerical columns
rownames(dat) <- rho_autop_sin[,1]
dat=t(dat)
b<-hclust(dist(t(dat)), method="ward.D2")

pdf("_Graph/heatmap_sinAUfull.99abs.pdf")
heatmap3(sqrt(dat),
          Colv=as.dendrogram(b),     # Reordered cols
          Rowv=as.dendrogram(a),     # Reordered rows
          method="ward.D2",
          scale="none", balanceColor =F,
         col = colorRampPalette(c("white", "darkblue"))(1024), cexCol = 0.4,cexRow =1.3)
dev.off()

snpsplit_ausin_all = as.data.frame(cutree(b,4))
snpsplit_ausin_all$contig = rownames(snpsplit_ausin_all)
colnames(snpsplit_ausin_all) = c( "cluster","contig")
```


```{r}
dat <- (abs(rho_natop_sin[,c(3:4, 6:25)]))  # numerical columns
rownames(dat) <- rho_natop_sin[,1]
dat=t(dat)

superheat((dat), row.dendrogram = T, col.dendrogram = T,  scale=F , heat.pal=(c("white", "darkblue")))
```

Galaxy biplots for outliers in all ranges
---
```{r}
rho_natop_cl = left_join(rho_natop,snpsplit_na_all, by="SNPID")
rho_eutop_cl = left_join(rho_eutop,snpsplit_eu_all, by="SNPID")
rho_autop_cl = left_join(rho_autop,snpsplit_au_all, by="SNPID")

rho_NAEU = inner_join(rho_na, rho_eu, by="SNPID")
rho_NAAU = inner_join(rho_na, rho_au, by="SNPID")
rho_EUAU = inner_join(rho_eu, rho_au, by="SNPID")
rho_NAEUtop = inner_join(rho_natop_cl, rho_eutop_cl, by="SNPID")
rho_NAAUtop = inner_join(rho_natop_cl, rho_autop_cl, by="SNPID")
rho_EUAUtop = inner_join(rho_eutop_cl, rho_autop_cl, by="SNPID")
```

Mean annual temperature versus precipitation
```{r}
jpeg("_Graph/Galbiplot_NA_MATMAP_cluster.jpg", height=9, width=9, units="cm", res=300)
 ggplot(rho_na, aes(x=bio1_all, y=bio12_all)) + geom_point(colour='grey', size=0.5) +theme_bw() +
  xlab("Tmean")+ ylab("Pmean")+stat_ellipse() +
  geom_point(data = rho_natop_cl,aes(x=bio1_all, y=bio12_all), colour=rho_natop_cl$cluster, size=2) +
  geom_text_repel(data=rho_natop_cl, label=rho_natop_cl$contig, size=3.5) 
dev.off()

jpeg("_Graph/Galbiplot_EU_MATMAP_cluster.jpg", height=9, width=9, units="cm", res=300)
 ggplot(rho_eu, aes(x=bio1_all, y=bio12_all)) + geom_point(colour='grey', size=0.5) +theme_bw() +
  xlab("Tmean")+ ylab("Pmean")+stat_ellipse() +
  geom_point(data = rho_eutop_cl,aes(x=bio1_all, y=bio12_all), colour=rho_eutop_cl$cluster, size=2) +
  geom_text_repel(data=rho_eutop_cl, label=rho_eutop_cl$contig, size=3.5) 
dev.off()

jpeg("_Graph/Galbiplot_AU_MATMAP_cluster.jpg", height=9, width=9, units="cm", res=300)
 ggplot(rho_au, aes(x=bio1_all, y=bio12_all)) + geom_point(colour='grey', size=0.5) +theme_bw() +
  xlab("Tmean")+ ylab("Pmean")+stat_ellipse() +
  geom_point(data = rho_autop_cl,aes(x=bio1_all, y=bio12_all), colour=rho_autop_cl$cluster, size=2) +
  geom_text_repel(data=rho_autop_cl, label=rho_autop_cl$contig, size=3.5) 
dev.off()

```
But what are the same SNPs? Labels with snp names?


Galaxy biplots for outliers in all ranges
---
```{r}
snpsplit_nasin_all$contig = as.numeric(snpsplit_nasin_all$contig)
snpsplit_eusin_all$contig = as.numeric(snpsplit_eusin_all$contig)
snpsplit_ausin_all$contig = as.numeric(snpsplit_ausin_all$contig)

rho_natopsin_cl = left_join(rho_natop_sin,snpsplit_nasin_all, by="contig")
rho_eutopsin_cl = left_join(rho_eutop_sin,snpsplit_eusin_all, by="contig")
rho_autopsin_cl = left_join(rho_autop_sin,snpsplit_ausin_all, by="contig")
```

Mean annual temperature versus precipitation
```{r}
jpeg("_Graph/Galbiplot_NAsin_MATMAP_cluster.jpg", height=9, width=9, units="cm", res=300)
 ggplot(rho_na, aes(x=Tmean, y=Pmean)) + geom_point(colour='grey', size=0.5) +theme_bw() +
  xlab("Tmean")+ ylab("Pmean")+stat_ellipse() +
  geom_point(data = rho_natopsin_cl,aes(x=Tmean, y=Pmean), colour=rho_natopsin_cl$cluster, size=2)  +  geom_text_repel(data=rho_natopsin_cl, label=rho_natopsin_cl$contig, size=3.5) 
dev.off()

jpeg("_Graph/Galbiplot_EUsin_MATMAP_cluster.jpg", height=9, width=9, units="cm", res=300)
 ggplot(rho_eu, aes(x=Tmean, y=Pmean)) + geom_point(colour='grey', size=0.5) +theme_bw() +
  xlab("Tmean")+ ylab("Pmean")+stat_ellipse() +
  geom_point(data = rho_eutopsin_cl,aes(x=Tmean, y=Pmean), colour=rho_eutopsin_cl$cluster, size=2)+  geom_text_repel(data=rho_eutopsin_cl, label=rho_eutopsin_cl$contig, size=3.5) 
dev.off()

jpeg("_Graph/Galbiplot_AUsin_MATMAP_cluster.jpg", height=9, width=9, units="cm", res=300)
 ggplot(rho_au, aes(x=Tmean, y=Pmean)) + geom_point(colour='grey', size=0.5) +theme_bw() +
  xlab("Tmean")+ ylab("Pmean")+stat_ellipse() +
  geom_point(data = rho_autopsin_cl,aes(x=Tmean, y=Pmean), colour=rho_autopsin_cl$cluster, size=2) +  geom_text_repel(data=rho_autopsin_cl, label=rho_autopsin_cl$contig, size=3.5) 
dev.off()
```


---

Old
===

```{r}
jpeg("_Graph/Galbiplot_NAEU_MAT_cluster.jpg", height=3, width = 3)
ggplot(rho_NAEU, aes(x=bio1_all.x, y=bio1_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUtop,aes(x=bio1_all.x, y=bio1_all.y), colour=rho_NAEUtop$cluster.x, size=5) + geom_point(data = rho_NAEUtop,aes(x=bio1_all.x, y=bio1_all.y),colour=(rho_NAEUtop$cluster.y), size=3)+ theme_bw() + xlab("SNP-MAT associations (Spearman's rho within Europe")+ ylab("SNP-MAT associations (Spearman's rho within Europe") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) 

jpeg("_Graph/Galbiplot_NAEU_Lat_cluster.jpg")
ggplot(rho_NAEU, aes(x=Latitude_all.x, y=Latitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUtop,aes(x=Latitude_all.x, y=Latitude_all.y), colour=rho_NAEUtop$cluster.x, size=5) + geom_point(data = rho_NAEUtop,aes(x=Latitude_all.x, y=Latitude_all.y),colour=(rho_NAEUtop$cluster.y), size=3)+ theme_bw() + xlab("SNP-Lat associations (Spearman's rho within Europe")+ ylab("SNP-Lat associations (Spearman's rho within Europe") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) 

jpeg("_Graph/Galbiplot_NAEU_Pseas_cluster.jpg")
ggplot(rho_NAEU, aes(x=bio15_all.x, y=bio15_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUtop,aes(x=bio15_all.x, y=bio15_all.y), colour=rho_NAEUtop$cluster.x, size=5) + geom_point(data = rho_NAEUtop,aes(x=bio15_all.x, y=bio15_all.y),colour=(rho_NAEUtop$cluster.y), size=3)+ theme_bw() + xlab("SNP-Pseas associations (Spearman's rho within Europe")+ ylab("SNP-Pseas associations (Spearman's rho within Europe") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) 
```


Galaxy biplot based on environmental variables
---

Get overlapping SNPS
```{r}
topSNP_NAEU = inner_join(toptopSNP_NA, toptopSNP_EU, by="windIDenv")
topSNP_NAEU$SNPID = topSNP_NAEU$SNPID.x
topSNP_NAEU_r <- ddply(topSNP_NAEU, ~windID.x, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_NAEU_r_alt <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="alt_all"),]
topSNP_NAEU_r_lat <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="Latitude_all"),]
topSNP_NAEU_r_lon <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="Longitude_all"),]
topSNP_NAEU_r_bio1 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio1_all"),]
topSNP_NAEU_r_bio2 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio2_all"),]
topSNP_NAEU_r_bio15 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio15_all"),]
topSNP_NAEU_r_bio17 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio17_all"),]
topSNP_NAEU_r_bio8 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio8_all"),]
topSNP_NAEU_r_bio12 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio12_all"),]
topSNP_NAEU_r_bio4 <- topSNP_NAEU_r[which(topSNP_NAEU_r$envvar.x=="bio4_all"),]

rho_NAEU = inner_join(rho_na, rho_eu, by="SNPID")
rho_NAEUto_alt = semi_join(rho_NAEU, topSNP_NAEU_r_alt, by="SNPID")
rho_NAEUto_lon = semi_join(rho_NAEU, topSNP_NAEU_r_lon, by="SNPID")
rho_NAEUto_lat = semi_join(rho_NAEU, topSNP_NAEU_r_lat, by="SNPID")
rho_NAEUto_bio1 = semi_join(rho_NAEU, topSNP_NAEU_r_bio1, by="SNPID")
rho_NAEUto_bio2 = semi_join(rho_NAEU, topSNP_NAEU_r_bio2, by="SNPID")
rho_NAEUto_bio4 = semi_join(rho_NAEU, topSNP_NAEU_r_bio4, by="SNPID")
rho_NAEUto_bio8 = semi_join(rho_NAEU, topSNP_NAEU_r_bio8, by="SNPID")
rho_NAEUto_bio12 = semi_join(rho_NAEU, topSNP_NAEU_r_bio12, by="SNPID")
rho_NAEUto_bio15 = semi_join(rho_NAEU, topSNP_NAEU_r_bio15, by="SNPID")
rho_NAEUto_bio17 = semi_join(rho_NAEU, topSNP_NAEU_r_bio17, by="SNPID")

topSNP_NAAU = inner_join(toptopSNP_NA, toptopSNP_EU, by="windIDenv")
topSNP_NAAU$SNPID = topSNP_NAAU$SNPID.x
topSNP_NAAU_r <- ddply(topSNP_NAAU, ~windID.x, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_NAAU_r_alt <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="alt_all"),]
topSNP_NAAU_r_lat <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="Latitude_all"),]
topSNP_NAAU_r_lon <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="Longitude_all"),]
topSNP_NAAU_r_bio1 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio1_all"),]
topSNP_NAAU_r_bio2 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio2_all"),]
topSNP_NAAU_r_bio15 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio15_all"),]
topSNP_NAAU_r_bio17 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio17_all"),]
topSNP_NAAU_r_bio8 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio8_all"),]
topSNP_NAAU_r_bio12 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio12_all"),]
topSNP_NAAU_r_bio4 <- topSNP_NAAU_r[which(topSNP_NAAU_r$envvar.x=="bio4_all"),]

rho_NAAU = inner_join(rho_na, rho_eu, by="SNPID")
rho_NAAUto_alt = semi_join(rho_NAAU, topSNP_NAAU_r_alt, by="SNPID")
rho_NAAUto_lon = semi_join(rho_NAAU, topSNP_NAAU_r_lon, by="SNPID")
rho_NAAUto_lat = semi_join(rho_NAAU, topSNP_NAAU_r_lat, by="SNPID")
rho_NAAUto_bio1 = semi_join(rho_NAAU, topSNP_NAAU_r_bio1, by="SNPID")
rho_NAAUto_bio2 = semi_join(rho_NAAU, topSNP_NAAU_r_bio2, by="SNPID")
rho_NAAUto_bio4 = semi_join(rho_NAAU, topSNP_NAAU_r_bio4, by="SNPID")
rho_NAAUto_bio8 = semi_join(rho_NAAU, topSNP_NAAU_r_bio8, by="SNPID")
rho_NAAUto_bio12 = semi_join(rho_NAAU, topSNP_NAAU_r_bio12, by="SNPID")
rho_NAAUto_bio15 = semi_join(rho_NAAU, topSNP_NAAU_r_bio15, by="SNPID")
rho_NAAUto_bio17 = semi_join(rho_NAAU, topSNP_NAAU_r_bio17, by="SNPID")

topSNP_EUAU = inner_join(toptopSNP_NA, toptopSNP_EU, by="windIDenv")
topSNP_EUAU$SNPID = topSNP_EUAU$SNPID.x
topSNP_EUAU_r <- ddply(topSNP_EUAU, ~windID.x, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_EUAU_r_alt <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="alt_all"),]
topSNP_EUAU_r_lat <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="Latitude_all"),]
topSNP_EUAU_r_lon <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="Longitude_all"),]
topSNP_EUAU_r_bio1 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio1_all"),]
topSNP_EUAU_r_bio2 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio2_all"),]
topSNP_EUAU_r_bio15 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio15_all"),]
topSNP_EUAU_r_bio17 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio17_all"),]
topSNP_EUAU_r_bio8 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio8_all"),]
topSNP_EUAU_r_bio12 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio12_all"),]
topSNP_EUAU_r_bio4 <- topSNP_EUAU_r[which(topSNP_EUAU_r$envvar.x=="bio4_all"),]

rho_EUAU = inner_join(rho_na, rho_eu, by="SNPID")
rho_EUAUto_alt = semi_join(rho_EUAU, topSNP_EUAU_r_alt, by="SNPID")
rho_EUAUto_lon = semi_join(rho_EUAU, topSNP_EUAU_r_lon, by="SNPID")
rho_EUAUto_lat = semi_join(rho_EUAU, topSNP_EUAU_r_lat, by="SNPID")
rho_EUAUto_bio1 = semi_join(rho_EUAU, topSNP_EUAU_r_bio1, by="SNPID")
rho_EUAUto_bio2 = semi_join(rho_EUAU, topSNP_EUAU_r_bio2, by="SNPID")
rho_EUAUto_bio4 = semi_join(rho_EUAU, topSNP_EUAU_r_bio4, by="SNPID")
rho_EUAUto_bio8 = semi_join(rho_EUAU, topSNP_EUAU_r_bio8, by="SNPID")
rho_EUAUto_bio12 = semi_join(rho_EUAU, topSNP_EUAU_r_bio12, by="SNPID")
rho_EUAUto_bio15 = semi_join(rho_EUAU, topSNP_EUAU_r_bio15, by="SNPID")
rho_EUAUto_bio17 = semi_join(rho_EUAU, topSNP_EUAU_r_bio17, by="SNPID")
```

```{r}
jpeg("_Graph/Galbiplot_NAEU_MAT.jpg")
ggplot(rho_NAEU, aes(x=bio1_all.x, y=bio1_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAEU_MAP.jpg")
ggplot(rho_NAEU, aes(x=bio12_all.x, y=bio12_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAEU_Lon.jpg")
ggplot(rho_NAEU, aes(x=Longitude_all.x, y=Longitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAEU_MQPdry.jpg")
ggplot(rho_NAEU, aes(x=bio17_all.x, y=bio17_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAEU_LAT.jpg")
ggplot(rho_NAEU, aes(x=Latitude_all.x, y=Latitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAEU_ALT.jpg")
ggplot(rho_NAEU, aes(x=alt_all.x, y=alt_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAEU_PSEAS.jpg")
ggplot(rho_NAEU, aes(x=bio15_all.x, y=bio15_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAEUto_lat, colour="red") + geom_point(data = rho_NAEUto_lon, colour="olivedrab")  + geom_point(data = rho_NAEUto_bio15, colour="blue")  +geom_point(data = rho_NAEUto_bio4, colour="orange") + geom_point(data = rho_NAEUto_alt, colour="black") +geom_point(data = rho_NAEUto_bio2, colour="purple") +geom_point(data = rho_NAEUto_bio17, colour="pink")+geom_point(data = rho_NAEUto_bio8, colour="chartreuse") +geom_point(data = rho_NAEUto_bio1, colour="yellow")+geom_point(data = rho_NAEUto_bio12, colour="brown")+geom_point(data = rho_NAEUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()
```

```{r}
jpeg("_Graph/Galbiplot_NAAU_MAT.jpg")
ggplot(rho_NAAU, aes(x=bio1_all.x, y=bio1_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAAU_MAP.jpg")
ggplot(rho_NAAU, aes(x=bio12_all.x, y=bio12_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAAU_Lon.jpg")
ggplot(rho_NAAU, aes(x=Longitude_all.x, y=Longitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAAU_MQPdry.jpg")
ggplot(rho_NAAU, aes(x=bio17_all.x, y=bio17_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAAU_LAT.jpg")
ggplot(rho_NAAU, aes(x=Latitude_all.x, y=Latitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAAU_ALT.jpg")
ggplot(rho_NAAU, aes(x=alt_all.x, y=alt_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NAAU_PSEAS.jpg")
ggplot(rho_NAAU, aes(x=bio15_all.x, y=bio15_all.y)) + geom_point(colour='grey') + geom_point(data = rho_NAAUto_lat, colour="red") + geom_point(data = rho_NAAUto_lon, colour="olivedrab")  + geom_point(data = rho_NAAUto_bio15, colour="blue")  +geom_point(data = rho_NAAUto_bio4, colour="orange") + geom_point(data = rho_NAAUto_alt, colour="black") +geom_point(data = rho_NAAUto_bio2, colour="purple") +geom_point(data = rho_NAAUto_bio17, colour="pink")+geom_point(data = rho_NAAUto_bio8, colour="chartreuse") +geom_point(data = rho_NAAUto_bio1, colour="yellow")+geom_point(data = rho_NAAUto_bio12, colour="brown")+geom_point(data = rho_NAAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()
```

```{r}
jpeg("_Graph/Galbiplot_EUAU_MAT.jpg")
ggplot(rho_EUAU, aes(x=bio1_all.x, y=bio1_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_EUAU_MAP.jpg")
ggplot(rho_EUAU, aes(x=bio12_all.x, y=bio12_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_EUAU_Lon.jpg")
ggplot(rho_EUAU, aes(x=Longitude_all.x, y=Longitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_EUAU_MQPdry.jpg")
ggplot(rho_EUAU, aes(x=bio17_all.x, y=bio17_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_EUAU_LAT.jpg")
ggplot(rho_EUAU, aes(x=Latitude_all.x, y=Latitude_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_EUAU_ALT.jpg")
ggplot(rho_EUAU, aes(x=alt_all.x, y=alt_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_EUAU_PSEAS.jpg")
ggplot(rho_EUAU, aes(x=bio15_all.x, y=bio15_all.y)) + geom_point(colour='grey') + geom_point(data = rho_EUAUto_lat, colour="red") + geom_point(data = rho_EUAUto_lon, colour="olivedrab")  + geom_point(data = rho_EUAUto_bio15, colour="blue")  +geom_point(data = rho_EUAUto_bio4, colour="orange") + geom_point(data = rho_EUAUto_alt, colour="black") +geom_point(data = rho_EUAUto_bio2, colour="purple") +geom_point(data = rho_EUAUto_bio17, colour="pink")+geom_point(data = rho_EUAUto_bio8, colour="chartreuse") +geom_point(data = rho_EUAUto_bio1, colour="yellow")+geom_point(data = rho_EUAUto_bio12, colour="brown")+geom_point(data = rho_EUAUto_bio1, colour="yellow")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()
```


old
---
```{r}
topSNP_eu_bio1 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio1_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio2 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio2_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio3 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio3_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio4 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio4_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio5 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio5_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio6 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio6_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio7 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio7_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio8 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio8_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio9 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio9_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio10 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio10_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio11 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio11_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio12 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio12_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio13 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio13_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio14 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio14_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio15 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio15_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio16<- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio16_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio17 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio17_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio18 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio18_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_bio19 <- ddply(topSNP_eu[which(topSNP_eu$envvar=="bio19_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_lat <- ddply(topSNP_eu[which(topSNP_eu$envvar=="Latitude_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_lon <- ddply(topSNP_eu[which(topSNP_eu$envvar=="Longitude_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_alt <- ddply(topSNP_eu[which(topSNP_eu$envvar=="alt_all"&topSNP_eu$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})

toptopSNP_eu = rbind(topSNP_eu_bio1, topSNP_eu_bio2, topSNP_eu_bio3, topSNP_eu_bio4, topSNP_eu_bio5, topSNP_eu_bio6, topSNP_eu_bio7, topSNP_eu_bio8, topSNP_eu_bio9, topSNP_eu_bio10, topSNP_eu_bio11, topSNP_eu_bio12, topSNP_eu_bio13, topSNP_eu_bio14, topSNP_eu_bio15, topSNP_eu_bio16, topSNP_eu_bio17, topSNP_eu_bio18, topSNP_eu_bio19, topSNP_eu_alt, topSNP_eu_lat, topSNP_eu_lon)

rho_eutop <- semi_join(rho_eu, toptopSNP_eu, by="SNPID")
#rho_eutopm =melt(rho_eutop, id=c("contig","location"))
rho_euto_bio1 = semi_join(rho_eu, topSNP_eu_bio1, by="SNPID")
rho_euto_bio2 = semi_join(rho_eu, topSNP_eu_bio2, by="SNPID")
rho_euto_bio3 = semi_join(rho_eu, topSNP_eu_bio3, by="SNPID")
rho_euto_bio4 = semi_join(rho_eu, topSNP_eu_bio4, by="SNPID")
rho_euto_bio5 = semi_join(rho_eu, topSNP_eu_bio5, by="SNPID")
rho_euto_bio6 = semi_join(rho_eu, topSNP_eu_bio6, by="SNPID")
rho_euto_bio7 = semi_join(rho_eu, topSNP_eu_bio7, by="SNPID")
rho_euto_bio8 = semi_join(rho_eu, topSNP_eu_bio8, by="SNPID")
rho_euto_bio9 = semi_join(rho_eu, topSNP_eu_bio9, by="SNPID")
rho_euto_bio10 = semi_join(rho_eu, topSNP_eu_bio10, by="SNPID")
rho_euto_bio11 = semi_join(rho_eu, topSNP_eu_bio11, by="SNPID")
rho_euto_bio12 = semi_join(rho_eu, topSNP_eu_bio12, by="SNPID")
rho_euto_bio13 = semi_join(rho_eu, topSNP_eu_bio13, by="SNPID")
rho_euto_bio14 = semi_join(rho_eu, topSNP_eu_bio14, by="SNPID")
rho_euto_bio15 = semi_join(rho_eu, topSNP_eu_bio15, by="SNPID")
rho_euto_bio16 = semi_join(rho_eu, topSNP_eu_bio16, by="SNPID")
rho_euto_bio17 = semi_join(rho_eu, topSNP_eu_bio17, by="SNPID")
rho_euto_bio18 = semi_join(rho_eu, topSNP_eu_bio18, by="SNPID")
rho_euto_bio19 = semi_join(rho_eu, topSNP_eu_bio19, by="SNPID")
rho_euto_lon = semi_join(rho_eu, topSNP_eu_lon, by="SNPID")
rho_euto_alt = semi_join(rho_eu, topSNP_eu_alt, by="SNPID")
rho_euto_lat = semi_join(rho_eu, topSNP_eu_lat, by="SNPID")
```


```{r}
topSNP_na_bio1 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio1_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio2 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio2_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio3 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio3_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio4 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio4_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio5 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio5_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio6 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio6_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio7 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio7_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio8 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio8_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio9 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio9_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio10 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio10_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio11 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio11_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio12 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio12_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio13 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio13_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio14 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio14_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio15 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio15_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio16<- ddply(topSNP_na[which(topSNP_na$envvar=="bio16_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio17 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio17_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio18 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio18_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_bio19 <- ddply(topSNP_na[which(topSNP_na$envvar=="bio19_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_lat <- ddply(topSNP_na[which(topSNP_na$envvar=="Latitude_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_lon <- ddply(topSNP_na[which(topSNP_na$envvar=="Longitude_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_alt <- ddply(topSNP_na[which(topSNP_na$envvar=="alt_all"&topSNP_na$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})

toptopSNP_na = rbind(topSNP_na_bio1, topSNP_na_bio2, topSNP_na_bio3, topSNP_na_bio4, topSNP_na_bio5, topSNP_na_bio6, topSNP_na_bio7, topSNP_na_bio8, topSNP_na_bio9, topSNP_na_bio10, topSNP_na_bio11, topSNP_na_bio12, topSNP_na_bio13, topSNP_na_bio14, topSNP_na_bio15, topSNP_na_bio16, topSNP_na_bio17, topSNP_na_bio18, topSNP_na_bio19, topSNP_na_alt, topSNP_na_lat, topSNP_na_lon)

rho_natop <- semi_join(rho_na, toptopSNP_na, by="SNPID")
#rho_natopm =melt(rho_natop, id=c("contig","location"))
rho_nato_bio1 = semi_join(rho_na, topSNP_na_bio1, by="SNPID")
rho_nato_bio2 = semi_join(rho_na, topSNP_na_bio2, by="SNPID")
rho_nato_bio3 = semi_join(rho_na, topSNP_na_bio3, by="SNPID")
rho_nato_bio4 = semi_join(rho_na, topSNP_na_bio4, by="SNPID")
rho_nato_bio5 = semi_join(rho_na, topSNP_na_bio5, by="SNPID")
rho_nato_bio6 = semi_join(rho_na, topSNP_na_bio6, by="SNPID")
rho_nato_bio7 = semi_join(rho_na, topSNP_na_bio7, by="SNPID")
rho_nato_bio8 = semi_join(rho_na, topSNP_na_bio8, by="SNPID")
rho_nato_bio9 = semi_join(rho_na, topSNP_na_bio9, by="SNPID")
rho_nato_bio10 = semi_join(rho_na, topSNP_na_bio10, by="SNPID")
rho_nato_bio11 = semi_join(rho_na, topSNP_na_bio11, by="SNPID")
rho_nato_bio12 = semi_join(rho_na, topSNP_na_bio12, by="SNPID")
rho_nato_bio13 = semi_join(rho_na, topSNP_na_bio13, by="SNPID")
rho_nato_bio14 = semi_join(rho_na, topSNP_na_bio14, by="SNPID")
rho_nato_bio15 = semi_join(rho_na, topSNP_na_bio15, by="SNPID")
rho_nato_bio16 = semi_join(rho_na, topSNP_na_bio16, by="SNPID")
rho_nato_bio17 = semi_join(rho_na, topSNP_na_bio17, by="SNPID")
rho_nato_bio18 = semi_join(rho_na, topSNP_na_bio18, by="SNPID")
rho_nato_bio19 = semi_join(rho_na, topSNP_na_bio19, by="SNPID")
rho_nato_lon = semi_join(rho_na, topSNP_na_lon, by="SNPID")
rho_nato_alt = semi_join(rho_na, topSNP_na_alt, by="SNPID")
rho_nato_lat = semi_join(rho_na, topSNP_na_lat, by="SNPID")
```

```{r}
topSNP_au_bio1 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio1_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio2 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio2_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio3 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio3_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio4 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio4_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio5 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio5_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio6 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio6_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio7 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio7_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio8 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio8_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio9 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio9_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio10 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio10_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio11 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio11_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio12 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio12_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio13 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio13_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio14 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio14_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio15 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio15_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio16<- ddply(topSNP_au[which(topSNP_au$envvar=="bio16_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio17 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio17_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio18 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio18_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_bio19 <- ddply(topSNP_au[which(topSNP_au$envvar=="bio19_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_lat <- ddply(topSNP_au[which(topSNP_au$envvar=="Latitude_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_lon <- ddply(topSNP_au[which(topSNP_au$envvar=="Longitude_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_alt <- ddply(topSNP_au[which(topSNP_au$envvar=="alt_all"&topSNP_au$rank>0.99),], ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})

toptopSNP_au = rbind(topSNP_au_bio1, topSNP_au_bio2, topSNP_au_bio3, topSNP_au_bio4, topSNP_au_bio5, topSNP_au_bio6, topSNP_au_bio7, topSNP_au_bio8, topSNP_au_bio9, topSNP_au_bio10, topSNP_au_bio11, topSNP_au_bio12, topSNP_au_bio13, topSNP_au_bio14, topSNP_au_bio15, topSNP_au_bio16, topSNP_au_bio17, topSNP_au_bio18, topSNP_au_bio19, topSNP_au_alt, topSNP_au_lat, topSNP_au_lon)

rho_autop <- semi_join(rho_au, toptopSNP_au, by="SNPID")
#rho_autopm =melt(rho_autop, id=c("contig","location"))
rho_auto_bio1 = semi_join(rho_au, topSNP_au_bio1, by="SNPID")
rho_auto_bio2 = semi_join(rho_au, topSNP_au_bio2, by="SNPID")
rho_auto_bio3 = semi_join(rho_au, topSNP_au_bio3, by="SNPID")
rho_auto_bio4 = semi_join(rho_au, topSNP_au_bio4, by="SNPID")
rho_auto_bio5 = semi_join(rho_au, topSNP_au_bio5, by="SNPID")
rho_auto_bio6 = semi_join(rho_au, topSNP_au_bio6, by="SNPID")
rho_auto_bio7 = semi_join(rho_au, topSNP_au_bio7, by="SNPID")
rho_auto_bio8 = semi_join(rho_au, topSNP_au_bio8, by="SNPID")
rho_auto_bio9 = semi_join(rho_au, topSNP_au_bio9, by="SNPID")
rho_auto_bio10 = semi_join(rho_au, topSNP_au_bio10, by="SNPID")
rho_auto_bio11 = semi_join(rho_au, topSNP_au_bio11, by="SNPID")
rho_auto_bio12 = semi_join(rho_au, topSNP_au_bio12, by="SNPID")
rho_auto_bio13 = semi_join(rho_au, topSNP_au_bio13, by="SNPID")
rho_auto_bio14 = semi_join(rho_au, topSNP_au_bio14, by="SNPID")
rho_auto_bio15 = semi_join(rho_au, topSNP_au_bio15, by="SNPID")
rho_auto_bio16 = semi_join(rho_au, topSNP_au_bio16, by="SNPID")
rho_auto_bio17 = semi_join(rho_au, topSNP_au_bio17, by="SNPID")
rho_auto_bio18 = semi_join(rho_au, topSNP_au_bio18, by="SNPID")
rho_auto_bio19 = semi_join(rho_au, topSNP_au_bio19, by="SNPID")
rho_auto_lon = semi_join(rho_au, topSNP_au_lon, by="SNPID")
rho_auto_alt = semi_join(rho_au, topSNP_au_alt, by="SNPID")
rho_auto_lat = semi_join(rho_au, topSNP_au_lat, by="SNPID")
```
---

```{r}
jpeg("_Graph/Galbiplot_NA_MATMAP.jpg")
ggplot(rho_na, aes(x=bio1_all, y=bio12_all)) + geom_point(colour='grey') + geom_point(data = rho_nato_lat, colour="red") + geom_point(data = rho_nato_lon, colour="olivedrab")  + geom_point(data = rho_nato_bio15, colour="blue")  +geom_point(data = rho_nato_bio4, colour="orange") + geom_point(data = rho_nato_alt, colour="black") +geom_point(data = rho_nato_bio2, colour="purple") +geom_point(data = rho_nato_bio17, colour="pink")+geom_point(data = rho_nato_bio8, colour="chartreuse") +geom_point(data = rho_nato_bio1, colour="yellow")+geom_point(data = rho_nato_bio12, colour="brown")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse() + xlim(-0.4,0.4) + ylim(-0.4,0.4) +theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NA_MATALT.jpg")
ggplot(rho_na, aes(x=bio1_all, y=alt_all)) + geom_point(colour='grey') + geom_point(data = rho_nato_lat, colour="red") + geom_point(data = rho_nato_lon, colour="olivedrab")  + geom_point(data = rho_nato_bio15, colour="blue")  +geom_point(data = rho_nato_bio4, colour="orange") + geom_point(data = rho_nato_alt, colour="black") +geom_point(data = rho_nato_bio2, colour="purple") +geom_point(data = rho_nato_bio17, colour="pink")+geom_point(data = rho_nato_bio8, colour="chartreuse")+geom_point(data = rho_nato_bio1, colour="yellow")+geom_point(data = rho_nato_bio12, colour="brown") +theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\naltitude") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_NA_MATLAT.jpg")
ggplot(rho_na, aes(x=bio1_all, y=Latitude_all)) + geom_point(colour='grey') +geom_point(data = rho_nato_lat, colour="red") + geom_point(data = rho_nato_lon, colour="olivedrab")  + geom_point(data = rho_nato_bio15, colour="blue")  +geom_point(data = rho_nato_bio4, colour="orange") + geom_point(data = rho_nato_alt, colour="black") +geom_point(data = rho_nato_bio2, colour="purple") +geom_point(data = rho_nato_bio17, colour="pink")+geom_point(data = rho_nato_bio8, colour="chartreuse")+geom_point(data = rho_nato_bio1, colour="yellow")+geom_point(data = rho_nato_bio12, colour="brown") +theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nlatitude") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_eu_MATMAP.jpg")
ggplot(rho_eu, aes(x=bio1_all, y=bio12_all)) + geom_point(colour='grey') + geom_point(data = rho_euto_lat, colour="red") + geom_point(data = rho_euto_lon, colour="olivedrab")  + geom_point(data = rho_euto_bio15, colour="blue")  +geom_point(data = rho_euto_bio4, colour="orange") + geom_point(data = rho_euto_alt, colour="black") +geom_point(data = rho_euto_bio2, colour="purple")+geom_point(data = rho_euto_bio1, colour="yellow") +geom_point(data = rho_euto_bio17, colour="pink")+geom_point(data = rho_euto_bio8, colour="chartreuse")+geom_point(data = rho_euto_bio12, colour="brown")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_eu_MATALT.jpg")
ggplot(rho_eu, aes(x=bio1_all, y=alt_all)) + geom_point(colour='grey') + geom_point(data = rho_euto_lat, colour="red") + geom_point(data = rho_euto_lon, colour="olivedrab")  + geom_point(data = rho_euto_bio15, colour="blue")  +geom_point(data = rho_euto_bio4, colour="orange") + geom_point(data = rho_euto_alt, colour="black") +geom_point(data = rho_euto_bio2, colour="purple") +geom_point(data = rho_euto_bio17, colour="pink")+geom_point(data = rho_euto_bio8, colour="chartreuse")+geom_point(data = rho_euto_bio1, colour="yellow")+geom_point(data = rho_euto_bio12, colour="brown") +theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\naltitude") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_eu_MATLAT.jpg")
ggplot(rho_eu, aes(x=bio1_all, y=Latitude_all)) + geom_point(colour='grey')  +geom_point(data = rho_euto_lat, colour="red")  + geom_point(data = rho_euto_lon, colour="olivedrab")  + geom_point(data = rho_euto_bio15, colour="blue")  +geom_point(data = rho_euto_bio4, colour="orange") + geom_point(data = rho_euto_alt, colour="black") +geom_point(data = rho_euto_bio2, colour="purple") +geom_point(data = rho_euto_bio17, colour="pink")+geom_point(data = rho_euto_bio8, colour="chartreuse")+geom_point(data = rho_euto_bio1, colour="yellow")+geom_point(data = rho_euto_bio12, colour="brown") +theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nlatitude") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_au_MATMAP.jpg")
ggplot(rho_au, aes(x=bio1_all, y=bio12_all)) + geom_point(colour='grey') + geom_point(data = rho_auto_lat, colour="red") + geom_point(data = rho_auto_lon, colour="olivedrab")  + geom_point(data = rho_auto_bio15, colour="blue")  +geom_point(data = rho_auto_bio4, colour="orange") + geom_point(data = rho_auto_alt, colour="black") +geom_point(data = rho_auto_bio2, colour="purple")+geom_point(data = rho_auto_bio1, colour="yellow") +geom_point(data = rho_auto_bio17, colour="pink")+geom_point(data = rho_auto_bio8, colour="chartreuse")+geom_point(data = rho_auto_bio12, colour="brown") + theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_au_MATALT.jpg")
ggplot(rho_au, aes(x=bio1_all, y=alt_all)) + geom_point(colour='grey') + geom_point(data = rho_auto_lat, colour="red") + geom_point(data = rho_auto_lon, colour="olivedrab")  + geom_point(data = rho_auto_bio15, colour="blue")  +geom_point(data = rho_auto_bio4, colour="orange") + geom_point(data = rho_auto_alt, colour="black") +geom_point(data = rho_auto_bio2, colour="purple") +geom_point(data = rho_auto_bio17, colour="pink")+geom_point(data = rho_auto_bio8, colour="chartreuse")+geom_point(data = rho_auto_bio12, colour="brown") +geom_point(data = rho_auto_bio1, colour="yellow") +theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\naltitude") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()

jpeg("_Graph/Galbiplot_au_MATLAT.jpg")
ggplot(rho_au, aes(x=bio1_all, y=Latitude_all)) + geom_point(colour='grey') +geom_point(data = rho_auto_lat, colour="red")  + geom_point(data = rho_auto_lon, colour="olivedrab")  + geom_point(data = rho_auto_bio15, colour="blue")  +geom_point(data = rho_auto_bio4, colour="orange") + geom_point(data = rho_auto_alt, colour="black") +geom_point(data = rho_auto_bio2, colour="purple") +geom_point(data = rho_auto_bio1, colour="yellow")+geom_point(data = rho_auto_bio17, colour="pink")+geom_point(data = rho_auto_bio8, colour="chartreuse")+geom_point(data = rho_auto_bio12, colour="brown") +theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nlatitude") +stat_ellipse()+ xlim(-0.4,0.4) + ylim(-0.4,0.4)+theme(axis.text.x = element_blank(), axis.text.y = element_blank(), axis.ticks = element_blank(),axis.title.x = element_blank(), axis.title.y = element_blank())
dev.off()
```

Compare galaxy biplots between ranges, not sure what this does
```{r}
rho_naeu = left_join(rho_na, rho_eu, by="SNPID")
rho_naeuto_lat = left_join(rho_nato_lat, rho_euto_lat, by="SNPID")
rho_naeuto_lon = left_join(rho_nato_lon, rho_euto_lon, by="SNPID")
rho_naeuto_bio15 = left_join(rho_nato_bio15, rho_euto_bio15, by="SNPID")

#pdf("_Graph/Galbiplot_NAEU_MATMAP.pdf")
ggplot(rho_naeu, aes(x=bio12_all.x, y=bio12_all.y)) + geom_point(colour='grey') + geom_point(data = rho_naeuto_lat, colour="red") + geom_point(data = rho_naeuto_lon, colour="olivedrab") + geom_point(data = rho_naeuto_bio15, colour="blue")
#+geom_point(data = rho_nato_bio4, colour="orange") + geom_point(data = rho_nato_alt, colour="black") +geom_point(data = rho_nato_bio2, colour="purple") +geom_point(data = rho_nato_bio17, colour="pink")+geom_point(data = rho_nato_bio8, colour="chartreuse")+ theme_bw() + xlab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean temperature") + ylab("Spearmans rho between population structure\ncorrected allele frequencies and\nannual mean precipitation") +stat_ellipse()
#dev.off()


```


All
```{r}
dat <- abs(rho_alltop[,c(3:25)])  # numerical columns
rownames(dat) <- rho_alltop[,26]
a<- hclust(dist(dat), method="ward.D2")
row.order <- hclust(dist(dat), method="ward.D2")$order # clustering
b<-hclust(dist(t(dat)), method="ward.D2")
col.order <- hclust(dist(t(dat)), method="ward.D2")$order
dat_new <- dat[row.order, col.order] # re-order matrix accoring to clustering

# Plot heatmap with heatmap.2
pdf("_Graph/heatmap_GLOBAL.pdf")
heatmap3(dat_new,                     # Tidy, normalised data
          Colv=as.dendrogram(b),     # Experiments clusters in cols
          Rowv=as.dendrogram(a),     # Protein clusters in rows
          density.info="histogram",   # Plot histogram of data and colour key
          trace="none",               # Turn of trace lines from heat map
          cexRow=0.5,cexCol=0.75
         ) 
dev.off()
```
#END SCRIPT#

