LD analyses
===  

> Script by Lotte van Boheemen  

```{r, include=F}
library("knitr")
library("dplyr")
library("plyr")
library("perm")
opts_knit$set(root.dir = '/Users/lottevanboheemen/Documents/Monash/PhD/Analyses/Data')
```

---  

Read in the data. Note that below LD is calculated within and not among scaffolds
```{r}
LDdata <- read.table("LD_results_out")
 colnames(LDdata) <- c("Scaf1","Loc1","Scaf2","Loc2","Corr")

LDdata$Dist <- abs(LDdata$Loc2-LDdata$Loc1)
```

Plot the results
```{r}
plot(LDdata$Corr~LDdata$Dist ,xlim=c(0,1000))
plot(abs(LDdata$Corr)~LDdata$Dist,xlim=c(0,1000))
plot(abs(LDdata$Corr)~log(LDdata$Dist), xlim=c(0,log(1000)))
m <- lm(Corr~log(Dist), data=LDdata)
visreg(m, xlim=c(0,1000))
abline(v=log(500)) # break-down of LD
LDdata$absCorr <- abs(LDdata$Corr)
```


Read in outlier file
```{r}
xtxna <- read.table("xtxna_1000_top99") # outlier window
xtxna$sub1 <- paste (xtxna$contig,xtxna$loc.start,sep = "__")
LDdata$sub1 <- paste(LDdata$Scaf1,LDdata$Loc1,sep="__")

cdata <- read.table("refgenome_data.txt",header=T) # reference genome data
xdat_na<-read.table("xtx_na",header=T) # xtx data native range
x_na<-left_join(xdat_na,cdata,by="contig") 
x_na$contig<-as.factor(x_na$contig)
x_na$sub1 <- paste(x_na$contig,x_na$location,sep="__")
x_na$myrank<-rank(x_na$xtx)/length(x_na$xtx)

xdat_eu <- read.table("xtx_eu.txt",header=T)
x_eu <- left_join(xdat_eu, cdata, by = "contig") # xtx data native range
x_eu$contig <- as.factor(x_eu$contig)
x_eu$sub1 <- paste(x_eu$contig,x_eu$location,sep="__")
x_eu$myrank <-rank(x_eu$xtx)/length(x_eu$xtx)

xdat_au <- read.table("xtx_au.txt",header=T)
x_au <- left_join(xdat_au, cdata, by = "contig") # xtx data native range
x_au$contig <- as.factor(x_au$contig)
x_au$sub1 <- paste(x_au$contig,x_au$location,sep="__")
x_au$myrank <-rank(x_au$xtx)/length(x_au$xtx)

joined <- inner_join(xtxna,LDdata,by="sub1")
joined2 <- inner_join(x_na,LDdata,by="sub1")
joined3 <- inner_join(LDdata,x_eu,by="sub1")
joined4 <- inner_join(LDdata,x_au,by="sub1")

```


Plot against distance, but have 2 colours, one = outlier, 1 is not

Explore if LD is higher around outliers
```{r}
plot(joined$win.mean~abs(joined$Corr)) # LD versus outlier window mean (native range)
str(joined2)
plot(abs(joined2$Corr)~joined2$Dist,col=ifelse(joined2$myrank>=0.95,"red","black")) # LD versus dist, coloured by xtx values in native range
plot(abs(joined2$Corr)~log(joined2$Dist),col=ifelse(joined2$myrank>=0.95,"red","black")) 
plot(abs(joined3$Corr)~joined3$Dist,col=ifelse(joined3$myrank>=0.95,"red","black"))# LD versus xtx values in european range
plot(abs(joined4$Corr)~joined4$Dist,col=ifelse(joined4$myrank>=0.95,"red","black"))# LD versus xtx values in Austlralian range
```


Read in the data. Note that below LD is calculated within and not among scaffolds
```{r}
LDdata <- read.table("LD_results_out")
colnames(LDdata) <- c("Scaf1","Loc1","Scaf2","Loc2","Corr")

LDdata$Dist <- abs(LDdata$Loc2-LDdata$Loc1)
LDdata$sub1 <- paste(LDdata$Scaf1,LDdata$Loc1,sep="__")
LDdata$sub2 <- paste(LDdata$Scaf2,LDdata$Loc2,sep="__")

snp_allwindow_na <- read.table("xtxna_1000_top99_SNPinfo",header = T)
snp_allwindow_na$sub1 <- paste(snp_allwindow_na$contig, snp_allwindow_na$location, sep="__")
snp_allwindow_na$sub2 <- snp_allwindow_na$sub1
snp_allwindow_na$windID <- paste(snp_allwindow_na$contig, snp_allwindow_na$win.start, snp_allwindow_na$win.end, sep="__") #all SNPs and window information they were allocated to

allwind_na <- read.table("sres1000_na99", header = T) #all windows
allwind_na$windID  <- paste(allwind_na$contig, allwind_na$win.start, allwind_na$win.end, sep="__")

super_na <- subset(allwind_na, topcan_q=="TRUE") #only select outlier windows
nonsuper_na <- subset(allwind_na, topcan_q=="FALSE") #all non-outlier windows
SNP_nonsuper_na <- inner_join(snp_allwindow_na, nonsuper_na, by = "windID") # all SNPs (+info) in a non-outlier window
SNP_super_na <- inner_join(snp_allwindow_na, super_na, by = "windID") # all SNPs (+info)  in a outlier window

SNP_nonsuper_na1 <- select(SNP_nonsuper_na, sub1, windID)
SNP_nonsuper_na2<- select(SNP_nonsuper_na, sub2, windID)

SNP_super_na1 <- select(SNP_super_na, sub1, windID)
SNP_super_na2<- select(SNP_super_na, sub2, windID)

temp <- inner_join(LDdata, SNP_nonsuper_na1, by ="sub1")
temp2<- inner_join(temp, SNP_nonsuper_na2, by ="sub2")
LD_nonoutwind_na <-temp2 [which(temp2$windID.x== temp2$windID.y),]

temp <- inner_join(LDdata, SNP_super_na1, by ="sub1")
temp2<- inner_join(temp, SNP_super_na2, by ="sub2")
LD_outwind_na <-temp2 [which(temp2$windID.x== temp2$windID.y),]
```

Plot distribution of correlation between SNPs within non-outlier and within outlier windows
```{r}
hist(LD_nonoutwind_na$Corr)
hist(LD_outwind_na$Corr)
```

Test if LD between SNPs within non-outlier vs outlier windows are different
```{r}
t.test(LD_nonoutwind_na$Corr, LD_outwind_na$Corr)

#doesn't look completely normal, as there is an overrepresentation of higher correlation values. Do a permutation test
Corr <- c(LD_outwind_na$Corr, LD_nonoutwind_na$Corr)
Corr_fac <- factor(rep(c("A", "B"), c(length(LD_outwind_na$Corr), length(LD_nonoutwind_na$Corr))))

perm.test1<- permTS(Corr ~ Corr_fac, alternative="greater", method="exact.mc", control=permControl(nmc=10^4-1))
perm.test1
```

Plot distribution of distance between SNPs within non-outlier and within outlier windows
```{r}
hist(LD_nonoutwind_na$Dist)
hist(log(LD_nonoutwind_na$Dist+1)) # does transformation normalizes the shape?
hist(LD_outwind_na$Dist)
hist(log(LD_outwind_na$Dist+1))
```

Test if distances between SNPs in outlier vs non-outlier windows are different
```{r}
wilcox.test(LD_outwind_na$Dist, LD_nonoutwind_na$Dist)
# above is not really allowed as the distributions don't have the same shape
# a permutation test would be better
Dist <- c(LD_outwind_na$Dist, LD_nonoutwind_na$Dist)
Dist_fac <- factor(rep(c("A", "B"), c(length(LD_outwind_na$Dist), length(LD_nonoutwind_na$Dist))))

perm.test2<- permTS(Dist ~ Dist_fac, alternative="greater", method="exact.mc", control=permControl(nmc=10^4-1))
perm.test2
```

Plot association between correlation and distance between SNPs within non-outlier and outlier windows
```{r}
plot(Corr ~ Dist, data= LD_outwind_na)
plot(Corr ~ Dist, data= LD_nonoutwind_na)
```


