---
output: html_document
editor_options: 
  chunk_output_type: console
---
**Select LDNA SNPs**
===


```{r}
library(dplyr)
library(plyr)
library(gdata)
library(igraph)
#devtools::install_github("petrikemppainen/LDna")
library(LDna)
```

Read in LD files
---
```{r}
ldna = read.table("na_ddGBS50.5.7_LD_infile")
ldeu = read.table("eu_ddGBS50.5.7_LD_infile")
ldau = read.table("au_ddGBS50.5.7_LD_infile")

ldna$SNPID = paste(ldna[,1],ldna[,2],sep = "__")
ldeu$SNPID = paste(ldeu[,1],ldeu[,2],sep = "__")
ldau$SNPID = paste(ldau[,1],ldau[,2],sep = "__")

```

Read data winds
---
```{r}
wind_na = read.table("bf_1000_99_allwind/sresbfall_na_1000_99_allwind.txt", header=T)
wind_eu = read.table("bf_1000_99_allwind/sresbfall_eu_1000_99_allwind.txt", header=T)
wind_au = read.table("bf_1000_99_allwind/sresbfall_au_1000_99_allwind.txt", header=T)
xwind_na = read.table("xtx_1000_99_allwind/sres1000-1000_na99_allwinds", header=T)
xwind_eu = read.table("xtx_1000_99_allwind/sres1000-1000_eu99_allwinds", header=T)
xwind_au = read.table("xtx_1000_99_allwind/sres1000-1000_au99_allwinds", header=T)
xwind_na$windID = paste0(xwind_na$contig, "__", xwind_na$win.start)
xwind_eu$windID = paste0(xwind_eu$contig, "__", xwind_eu$win.start)
xwind_au$windID = paste0(xwind_au$contig, "__", xwind_au$win.start)
wind_na$windID = paste0(wind_na$contig, "__", wind_na$win.start)
wind_eu$windID = paste0(wind_eu$contig, "__", wind_eu$win.start)
wind_au$windID = paste0(wind_au$contig, "__", wind_au$win.start)

allwindows1 = read.table("xtx_1000_99_allwind/xtxall_1000-1000_top99_allwinds", header=T)
allSNPS1 = read.table("xtx_1000_99_allwind/xtxall_1000-1000_top99_allwinds_SNPinfo", header=T)
allwindows = dplyr::select(allwindows1, contig, win.start)
allSNPS = dplyr::select(allSNPS1, contig, location, win.start)
allwindows$windID = paste0(allwindows$contig, "__", allwindows$win.start)

allSNPS$windID = paste0(allSNPS$contig, "__", allSNPS$win.start)
allSNPS$sub1 = paste0(allSNPS$contig, "__",allSNPS$location)
allSNPS$sub2 = paste0(allSNPS$contig, "__",allSNPS$location)

gcta_allwind = read.table("gcta_1000_allwind/gcta_all_1000-1000_top99_allwinds", header=T)
gcta_allwind$windID = paste0(gcta_allwind$contig,"__",gcta_allwind$win.start)
gcta = read.table("gcta_1000_allwind/gcta_1000_top99_supwind", header=T)
gcta$windID = paste0(gcta$contig,"__",gcta$win.start)
bfn_sup = read.table("bf_1000_99_allwind/mbf_1000_99_binnullW.txt", header=T)
xtxn_sup = read.table("xtx_1000_99_allwind/mxtx_1000_99_binnullW.txt", header=T)

bfn_sup_NA = bfn_sup %>% filter(Range=="nat"& qbinom=="yes") %>% droplevels()
bfn_sup_EU = bfn_sup %>% filter(Range=="EU"& qbinom=="yes") %>% droplevels()
bfn_sup_AU = bfn_sup %>% filter(Range=="AU"& qbinom=="yes") %>% droplevels()
xn_sup_NA = xtxn_sup %>% filter(Range=="nat"& binom=="yes") %>% droplevels()
xn_sup_EU = xtxn_sup %>% filter(Range=="EU"&binom=="yes") %>% droplevels()
xn_sup_AU = xtxn_sup %>% filter(Range=="EU"&binom=="yes") %>% droplevels()

xsnp_na <- read.table("xtx_1000_99_allwind/xtxna_1000-1000_top99_allwinds_SNPinfo",header = T)
xsnp_na$sub1 <- paste0(xsnp_na$contig,"__", xsnp_na$location)
xsnp_na$sub2 <- xsnp_na$sub1
xsnp_na$windID = paste0(xsnp_na$contig, "__", xsnp_na$win.start)

xsnp_eu <- read.table("xtx_1000_99_allwind/xtxeu_1000-1000_top99_allwinds_SNPinfo",header = T)
xsnp_eu$sub1 <- paste0(xsnp_eu$contig,"__", xsnp_eu$location)
xsnp_eu$sub2 <- xsnp_eu$sub1
xsnp_eu$windID = paste0(xsnp_eu$contig, "__", xsnp_eu$win.start)

xsnp_au <- read.table("xtx_1000_99_allwind/xtxau_1000-1000_top99_allwinds_SNPinfo",header = T)
xsnp_au$sub1 <- paste0(xsnp_au$contig,"__", xsnp_au$location)
xsnp_au$sub2 <- xsnp_au$sub1
xsnp_au$windID = paste0(xsnp_au$contig, "__", xsnp_au$win.start)

bsnp_na <- read.table("bf_1000_99_allwind/bfall_na_1000_bf99_allwinds_SNPinfo",header = T)
bsnp_na$sub1 <- paste0(bsnp_na$contig,"__", bsnp_na$location)
bsnp_na$sub2 <- bsnp_na$sub1
bsnp_na$windID = paste0(bsnp_na$contig, "__", bsnp_na$win.start)

bsnp_eu <- read.table("bf_1000_99_allwind/bfall_eu_1000_bf99_allwinds_SNPinfo",header = T)
bsnp_eu$sub1 <- paste0(bsnp_eu$contig,"__", bsnp_eu$location)
bsnp_eu$sub2 <- bsnp_eu$sub1
bsnp_eu$windID = paste0(bsnp_eu$contig, "__", bsnp_eu$win.start)

bsnp_au <- read.table("bf_1000_99_allwind/bfall_au_1000_bf99_allwinds_SNPinfo",header = T)
bsnp_au$sub1 <- paste0(bsnp_au$contig,"__", bsnp_au$location)
bsnp_au$sub2 <- bsnp_au$sub1
bsnp_au$windID = paste0(bsnp_au$contig, "__", bsnp_au$win.start)
```

Window has to have hit in at least 2 methods within a range (based on binom )
---
```{r}
bfn_sup_NAall = as.data.frame(levels(as.factor(bfn_sup_NA$windID)))
colnames(bfn_sup_NAall) = "windID"
xn_sup_NAall = dplyr::select(xn_sup_NA, windID)
bfn_sup_EUall = as.data.frame(levels(as.factor(bfn_sup_EU$windID)))
colnames(bfn_sup_EUall) = "windID"
xn_sup_EUall = dplyr::select(xn_sup_EU, windID)
bfn_sup_AUall = as.data.frame(levels(as.factor(bfn_sup_AU$windID)))
colnames(bfn_sup_AUall) = "windID"
xn_sup_AUall = dplyr::select(xn_sup_AU, windID)

bfxtx_sup_NA = intersect(xn_sup_NAall$windID, bfn_sup_NAall$windID)
bfxtx_sup_AU = intersect(xn_sup_AUall$windID, bfn_sup_AUall$windID)
bfxtx_sup_EU = intersect(xn_sup_EUall$windID, bfn_sup_EUall$windID)

gctawi = dplyr::select(gcta, windID)
bfgcta_sup_NA = intersect(gctawi$windID, bfn_sup_NAall$windID)
bfgcta_sup_EU = intersect(gctawi$windID, bfn_sup_EUall$windID)
bfgcta_sup_AU = intersect(gctawi$windID, bfn_sup_AUall$windID)

xgcta_sup_NA = intersect(gctawi$windID, xn_sup_NA$windID)
xgcta_sup_EU = intersect(gctawi$windID,  xn_sup_EU$windID)
xgcta_sup_AU = intersect(gctawi$windID, xn_sup_AU$windID)

bfxtxgcta_sup_NA = intersect(bfxtx_sup_NA, gctawi$windID) #conflicts with MASS
bfxtxgcta_sup_EU = intersect(bfxtx_sup_EU, gctawi$windID) 
bfxtxgcta_sup_AU = intersect(bfxtx_sup_AU, gctawi$windID) 

#found in at least 1 methods
wind_overlaptop_na_1m = rbind(bfn_sup_NAall, xn_sup_NAall)
wind_overlaptop_eu_1m = rbind(bfn_sup_EUall, xn_sup_EUall)
wind_overlaptop_au_1m = rbind(bfn_sup_AUall, xn_sup_AUall)

write.table(wind_overlaptop_na_1m, "wind_overlaptop_na_1m")
write.table(wind_overlaptop_eu_1m, "wind_overlaptop_eu_1m")
write.table(wind_overlaptop_au_1m, "wind_overlaptop_au_1m")

#found in at least 2 methods
#wind_overlaptop_na_2m = as.data.frame(c(bfxtx_sup_NA, bfgcta_sup_NA, xgcta_sup_NA))
#wind_overlaptop_eu_2m = as.data.frame(c(bfxtx_sup_EU, bfgcta_sup_EU, xgcta_sup_EU))
#wind_overlaptop_au_2m = as.data.frame(c(bfxtx_sup_AU, bfgcta_sup_AU,  xgcta_sup_AU))

wind_overlaptop_na_2m = as.data.frame(bfxtx_sup_NA)
wind_overlaptop_eu_2m = as.data.frame(bfxtx_sup_EU)
wind_overlaptop_au_2m = as.data.frame(bfxtx_sup_AU)

colnames(wind_overlaptop_na_2m) = "windID"
colnames(wind_overlaptop_eu_2m) = "windID"
colnames(wind_overlaptop_au_2m) = "windID"

write.table(wind_overlaptop_na_2m, "wind_overlaptop_na_2m")
write.table(wind_overlaptop_eu_2m, "wind_overlaptop_eu_2m")
write.table(wind_overlaptop_au_2m, "wind_overlaptop_au_2m")
```

Select random SNPs per window, NA
---
```{r}
SNP_na_50 = dplyr::select(ldna,SNPID)
SNP_na_50$sub1 = SNP_na_50$SNPID
bsnp_na1 = dplyr::select(bsnp_na, envvar, rank, sub1)

tmp = semi_join(xsnp_na, SNP_na_50, by="sub1") #retain all rows in a that have a match in b
topSNP_na_2m50_x = semi_join(tmp, wind_overlaptop_na_2m, by="windID") 
topSNP_na_2m50_xb1 = right_join(bsnp_na1, topSNP_na_2m50_x, by="sub1")
topSNP_na_2m50_xb2 = topSNP_na_2m50_xb1[which(topSNP_na_2m50_xb1$myrank>0.99&topSNP_na_2m50_xb1$rank>0.99),]
topSNP_na_2m50_xb3 = dplyr::select(topSNP_na_2m50_xb2, sub1, windID, contig, location)
topSNP_na_2m50_xb = dplyr::distinct(topSNP_na_2m50_xb3)

topSNP_na_2m50_rand =  ddply(topSNP_na_2m50_xb, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_na_2m50_rand$SNPID=topSNP_na_2m50_rand$sub1
write.table(topSNP_na_2m50_rand, "topSNP_na_2m50_rand")
#topSNP_na_2m_rand=read.table("topSNP_na_2m_rand", header=T)

#select non-outlier SNPs
notopSNP_na_2m50_x = anti_join(tmp, wind_overlaptop_na_1m, by = "windID") 
notopSNP_na_2m50_xb1 = right_join(bsnp_na1, notopSNP_na_2m50_x, by="sub1")

tmp = notopSNP_na_2m50_xb1[which(notopSNP_na_2m50_xb1$rank>0.99),]
tmp2 = dplyr::distinct(dplyr::select(tmp, sub1)) # SNPs that have BF>0.99 for any of the env
notopSNP_na_2m50_xb2 = dplyr::distinct( dplyr::select(notopSNP_na_2m50_xb1, sub1, windID, contig, location, myrank))
notopSNP_na_2m50_xb3 = anti_join(notopSNP_na_2m50_xb2, tmp2, by="sub1")
notopSNP_na_2m50_xb4 = notopSNP_na_2m50_xb3[which(notopSNP_na_2m50_xb3$myrank<0.99),]
notopSNP_na_2m50_xb = dplyr::select(notopSNP_na_2m50_xb4, sub1, windID, contig, location)

notopSNP_na_2m50_rand =  ddply(notopSNP_na_2m50_xb, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
notopSNP_na_2m50_rand$SNPID=notopSNP_na_2m50_rand$sub1

write.table(notopSNP_na_2m50_rand, "notopSNP_na_2m50_rand")
#topSNP_na_2m_rand=read.table("topSNP_na_2m_rand", header=T)

#select 1 window at random, used for statistical calculations?
topSNP_na_2m50_randSW =  ddply(topSNP_na_2m50_rand, ~contig, function(x){ndf <- x[sample(nrow(x),1), ]})
notopSNP_na_2m50_randSW =  ddply(notopSNP_na_2m50_rand, ~contig, function(x){ndf <- x[sample(nrow(x),1), ]})
topSNP_na_2m50_randSW$SNPID=topSNP_na_2m50_randSW$sub1
notopSNP_na_2m50_randSW$SNPID=notopSNP_na_2m50_randSW$sub1

write.table(topSNP_na_2m50_randSW, "topSNP_na_2m50_randSW")
write.table(notopSNP_na_2m50_randSW, "notopSNP_na_2m50_randSW")

randsel_na =gdata::combine(topSNP_na_2m50_rand,notopSNP_na_2m50_rand) #need to detach dplyr
LD_rand_na = semi_join(ldna, randsel_na, by="SNPID") 
LD_rand_top_na = semi_join(ldna, topSNP_na_2m50_rand, by="SNPID")
LD_rand_notop_na = semi_join(ldna, notopSNP_na_2m50_rand, by="SNPID")

randsel_na_SW =gdata::combine(topSNP_na_2m50_randSW,notopSNP_na_2m50_randSW) #need to detach dplyr
LD_rand_na_SW = semi_join(ldna, randsel_na, by="SNPID") 
LD_rand_top_na_SW = semi_join(ldna, topSNP_na_2m50_randSW, by="SNPID")
LD_rand_notop_na_SW = semi_join(ldna, notopSNP_na_2m50_randSW, by="SNPID")
```

Select random SNPs per window, EU
---
```{r}
SNP_eu_50 = dplyr::select(ldeu,SNPID)
SNP_eu_50$sub1 = SNP_eu_50$SNPID
bsnp_eu1 = dplyr::select(bsnp_eu, envvar, rank, sub1)

tmp = semi_join(xsnp_eu, SNP_eu_50, by="sub1") #retain all rows in a that have a match in b
topSNP_eu_2m50_x = semi_join(tmp, wind_overlaptop_eu_2m, by="windID") 
topSNP_eu_2m50_xb1 = right_join(bsnp_eu1, topSNP_eu_2m50_x, by="sub1")
topSNP_eu_2m50_xb2 = topSNP_eu_2m50_xb1[which(topSNP_eu_2m50_xb1$myrank>0.99&topSNP_eu_2m50_xb1$rank>0.99),]
topSNP_eu_2m50_xb3 = dplyr::select(topSNP_eu_2m50_xb2, sub1, windID, contig, location)
topSNP_eu_2m50_xb = dplyr::distinct(topSNP_eu_2m50_xb3)

topSNP_eu_2m50_rand =  ddply(topSNP_eu_2m50_xb, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_2m50_rand$SNPID=topSNP_eu_2m50_rand$sub1
write.table(topSNP_eu_2m50_rand, "topSNP_eu_2m50_rand")
#topSNP_eu_2m_rand=read.table("topSNP_eu_2m_rand", header=T)

#select non-outlier SNPs
notopSNP_eu_2m50_x = anti_join(tmp, wind_overlaptop_eu_1m, by = "windID") 
notopSNP_eu_2m50_xb1 = right_join(bsnp_eu1, notopSNP_eu_2m50_x, by="sub1")

tmp = notopSNP_eu_2m50_xb1[which(notopSNP_eu_2m50_xb1$rank>0.99),]
tmp2 = dplyr::distinct(dplyr::select(tmp, sub1)) # SNPs that have BF>0.99 for any of the env
notopSNP_eu_2m50_xb2 = dplyr::distinct( dplyr::select(notopSNP_eu_2m50_xb1, sub1, windID, contig, location, myrank))
notopSNP_eu_2m50_xb3 = anti_join(notopSNP_eu_2m50_xb2, tmp2, by="sub1")
notopSNP_eu_2m50_xb4 = notopSNP_eu_2m50_xb3[which(notopSNP_eu_2m50_xb3$myrank<0.99),]
notopSNP_eu_2m50_xb = dplyr::select(notopSNP_eu_2m50_xb4, sub1, windID, contig, location)

notopSNP_eu_2m50_rand =  ddply(notopSNP_eu_2m50_xb, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
notopSNP_eu_2m50_rand$SNPID=notopSNP_eu_2m50_rand$sub1

write.table(notopSNP_eu_2m50_rand, "notopSNP_eu_2m50_rand")
#topSNP_eu_2m_rand=read.table("topSNP_eu_2m_rand", header=T)

#select 1 window at random, used for statistical calculations?
topSNP_eu_2m50_randSW =  ddply(topSNP_eu_2m50_rand, ~contig, function(x){ndf <- x[sample(nrow(x),1), ]})
notopSNP_eu_2m50_randSW =  ddply(notopSNP_eu_2m50_rand, ~contig, function(x){ndf <- x[sample(nrow(x),1), ]})
topSNP_eu_2m50_randSW$SNPID=topSNP_eu_2m50_randSW$sub1
notopSNP_eu_2m50_randSW$SNPID=notopSNP_eu_2m50_randSW$sub1

write.table(topSNP_eu_2m50_randSW, "topSNP_eu_2m50_randSW")
write.table(notopSNP_eu_2m50_randSW, "notopSNP_eu_2m50_randSW")

randsel_eu =gdata::combine(topSNP_eu_2m50_rand,notopSNP_eu_2m50_rand) #need to detach dplyr
LD_rand_eu = semi_join(ldeu, randsel_eu, by="SNPID") 
LD_rand_top_eu = semi_join(ldeu, topSNP_eu_2m50_rand, by="SNPID")
LD_rand_notop_eu = semi_join(ldeu, notopSNP_eu_2m50_rand, by="SNPID")

randsel_eu_SW =gdata::combine(topSNP_eu_2m50_randSW,notopSNP_eu_2m50_randSW) #need to detach dplyr
LD_rand_eu_SW = semi_join(ldeu, randsel_eu, by="SNPID") 
LD_rand_top_eu_SW = semi_join(ldeu, topSNP_eu_2m50_randSW, by="SNPID")
LD_rand_notop_eu_SW = semi_join(ldeu, notopSNP_eu_2m50_randSW, by="SNPID")
```
Select random SNPs per window, AU
---
```{r}
SNP_au_50 = dplyr::select(ldau,SNPID)
SNP_au_50$sub1 = SNP_au_50$SNPID
bsnp_au1 = dplyr::select(bsnp_au, envvar, rank, sub1)

tmp = semi_join(xsnp_au, SNP_au_50, by="sub1") #retain all rows in a that have a match in b
topSNP_au_2m50_x = semi_join(tmp, wind_overlaptop_au_2m, by="windID") 
topSNP_au_2m50_xb1 = right_join(bsnp_au1, topSNP_au_2m50_x, by="sub1")
topSNP_au_2m50_xb2 = topSNP_au_2m50_xb1[which(topSNP_au_2m50_xb1$myrank>0.99&topSNP_au_2m50_xb1$rank>0.99),]
topSNP_au_2m50_xb3 = dplyr::select(topSNP_au_2m50_xb2, sub1, windID, contig, location)
topSNP_au_2m50_xb = dplyr::distinct(topSNP_au_2m50_xb3)

topSNP_au_2m50_rand =  ddply(topSNP_au_2m50_xb, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
topSNP_au_2m50_rand$SNPID=topSNP_au_2m50_rand$sub1
write.table(topSNP_au_2m50_rand, "topSNP_au_2m50_rand")
#topSNP_au_2m_rand=read.table("topSNP_au_2m_rand", header=T)

#select non-outlier SNPs
notopSNP_au_2m50_x = anti_join(tmp, wind_overlaptop_au_1m, by = "windID") 
notopSNP_au_2m50_xb1 = right_join(bsnp_au1, notopSNP_au_2m50_x, by="sub1")

tmp = notopSNP_au_2m50_xb1[which(notopSNP_au_2m50_xb1$rank>0.99),]
tmp2 = dplyr::distinct(dplyr::select(tmp, sub1)) # SNPs that have BF>0.99 for any of the env
notopSNP_au_2m50_xb2 = dplyr::distinct( dplyr::select(notopSNP_au_2m50_xb1, sub1, windID, contig, location, myrank))
notopSNP_au_2m50_xb3 = anti_join(notopSNP_au_2m50_xb2, tmp2, by="sub1")
notopSNP_au_2m50_xb4 = notopSNP_au_2m50_xb3[which(notopSNP_au_2m50_xb3$myrank<0.99),]
notopSNP_au_2m50_xb = dplyr::select(notopSNP_au_2m50_xb4, sub1, windID, contig, location)

notopSNP_au_2m50_rand =  ddply(notopSNP_au_2m50_xb, ~windID, function(x){ ndf <- x[sample(nrow(x),1), ]})
notopSNP_au_2m50_rand$SNPID=notopSNP_au_2m50_rand$sub1

write.table(notopSNP_au_2m50_rand, "notopSNP_au_2m50_rand")
#topSNP_au_2m_rand=read.table("topSNP_au_2m_rand", header=T)

#select 1 window at random, used for statistical calculations?
topSNP_au_2m50_randSW =  ddply(topSNP_au_2m50_rand, ~contig, function(x){ndf <- x[sample(nrow(x),1), ]})
notopSNP_au_2m50_randSW =  ddply(notopSNP_au_2m50_rand, ~contig, function(x){ndf <- x[sample(nrow(x),1), ]})
topSNP_au_2m50_randSW$SNPID=topSNP_au_2m50_randSW$sub1
notopSNP_au_2m50_randSW$SNPID=notopSNP_au_2m50_randSW$sub1

write.table(topSNP_au_2m50_randSW, "topSNP_au_2m50_randSW")
write.table(notopSNP_au_2m50_randSW, "notopSNP_au_2m50_randSW")

randsel_au =gdata::combine(topSNP_au_2m50_rand,notopSNP_au_2m50_rand) #need to detach dplyr
LD_rand_au = semi_join(ldau, randsel_au, by="SNPID") 
LD_rand_top_au = semi_join(ldau, topSNP_au_2m50_rand, by="SNPID")
LD_rand_notop_au = semi_join(ldau, notopSNP_au_2m50_rand, by="SNPID")

randsel_au_SW =gdata::combine(topSNP_au_2m50_randSW,notopSNP_au_2m50_randSW) #need to detach dplyr
LD_rand_au_SW = semi_join(ldau, randsel_au, by="SNPID") 
LD_rand_top_au_SW = semi_join(ldau, topSNP_au_2m50_randSW, by="SNPID")
LD_rand_notop_au_SW = semi_join(ldau, notopSNP_au_2m50_randSW, by="SNPID")
```
Calculate LD Based on SNPID NA
---
```{r}
#select SNPID and inds
input= LD_rand_na[,c(ncol(LD_rand_na),3:(ncol(LD_rand_na)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_na50_ddgbs_random_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID EU
---
```{r}
#select SNPID and inds
input= LD_rand_eu[,c(ncol(LD_rand_eu),3:(ncol(LD_rand_eu)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_eu50_ddgbs_random_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID AU
---
```{r}
#select SNPID and inds
input= LD_rand_au[,c(ncol(LD_rand_au),3:(ncol(LD_rand_au)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_au50_ddgbs_random_results_out_new",row.names = F, col.names = F, quote = F)
```




*NOT RUN YET*
===

Calculate LD Based on SNPID NA TOP
---
```{r}
#select SNPID and inds
input= LD_rand_top_na[,c(ncol(LD_rand_top_na),3:(ncol(LD_rand_top_na)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_na50_ddgbs_random_top_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID EU TOP
---
```{r}
#select SNPID and inds
input= LD_rand_top_eu[,c(ncol(LD_rand_top_eu),3:(ncol(LD_rand_top_eu)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_eu50_ddgbs_random_top_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID AU TOP
---
```{r}
#select SNPID and inds
input= LD_rand_top_au[,c(ncol(LD_rand_top_au),3:(ncol(LD_rand_top_au)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_au50_ddgbs_random_top_results_out_new",row.names = F, col.names = F, quote = F)
```



Calculate LD Based on SNPID NA NOTOP
---
```{r}
#select SNPID and inds
input= LD_rand_notop_na[,c(ncol(LD_rand_notop_na),3:(ncol(LD_rand_notop_na)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_na50_ddgbs_random_notop_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID EU NOTOP
---
```{r}
#select SNPID and inds
input= LD_rand_notop_eu[,c(ncol(LD_rand_notop_eu),3:(ncol(LD_rand_notop_eu)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_eu50_ddgbs_random_notop_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID AU NOTOP
---
```{r}
#select SNPID and inds
input= LD_rand_notop_au[,c(ncol(LD_rand_notop_au),3:(ncol(LD_rand_notop_au)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_au50_ddgbs_random_notop_results_out_new",row.names = F, col.names = F, quote = F)
```






Calculate LD Based on SNPID NA, 1 wind per contig
---
```{r}
#select SNPID and inds
input= LD_rand_na_SW[,c(ncol(LD_rand_na_SW),3:(ncol(LD_rand_na_SW)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_na50_ddgbs_randomSW_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID EU, 1 wind per contig
---
```{r}
#select SNPID and inds
input= LD_rand_eu_SW[,c(ncol(LD_rand_eu_SW),3:(ncol(LD_rand_eu_SW)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_eu50_ddgbs_randomSW_results_out_new",row.names = F, col.names = F, quote = F)
```

Calculate LD Based on SNPID AU, 1 wind per contig
---
```{r}
#select SNPID and inds
input= LD_rand_au_SW[,c(ncol(LD_rand_au_SW),3:(ncol(LD_rand_au_SW)-1))]

#this is the list of contigs and positions for each genotype
scafpos_good <- as.matrix (input[,1])

#this keeps the columns correct: 
input_good = data.matrix(input[,2:dim(input)[2]])

# Check if R is able to handle the amount of data 2^31 is the maximum extent of each dimension of an array R is able to handle
(nrow(input_good)^2)*3 < 2^31

results_out <- array (NA, c(((nrow (input_good))^2),3))
count = 0

#loop through focal snp
for (i in 1:nrow(input_good)){
#       loop through all SNPs
        for (j in 1:nrow (input_good)){
                if (i != j){
                count <- count + 1
                 results_out[count,3] <- cor (input_good[i,], input_good[j,], use = "pairwise.complete.obs")
                results_out [count,1] <- as.character (scafpos_good[i,1])
                results_out [count,2] <- as.character (scafpos_good[j,1])
                }
        }
}

write.table (results_out[1:count,],"LD_au50_ddgbs_randomSW_results_out_new",row.names = F, col.names = F, quote = F)
```




*NOT RUN YET*
===

##END SCRIPT##
===